---
title: "Getting Started with the Project 5004"
author: "Phoebe Fei"
date: "2/27/2022"
output: 
  html_document:
    toc: true
    toc_float: true
    toc_depth: 3
---

```{css style_settings, echo = FALSE}
blockquote { 
    font-size: 14px; 
    font-style: italic;
}
h1{
    font-size: 22px;
}
h2{
    font-size: 20px;
}
h3{
    font-size: 18px;
}
h4{
    font-size: 16px;
}

```

# Question 1: Downloading and Accessing Information of 1 fastq file

>Download at least one FASTQ file that you will be working with for your project. Document the following details: (2pt)

### a. Where did you get it from?

The data is accessible from NCBI SRA/GEO dataset under project number PRJNA386015. The link to the SRA run selector is https://www.ncbi.nlm.nih.gov/Traces/study/?acc=PRJNA386015&o=acc_s%3Aa. 

### b. What publication is it linked to?

Citation of this publication:
Bens, M., Szafranski, K., Holtze, S. et al. Naked mole-rat transcriptome signatures of socially suppressed sexual maturation and links of reproduction to aging. BMC Biol 16, 77 (2018). https://doi.org/10.1186/s12915-018-0546-z

### c. Who generated the data?

The data is generated by the authors of the above citation. Here is the contact information listed on GEO:
```
Submission date:	May 09, 2017
Last update date:	May 15, 2019
Contact name: Marco Groth
E-mail(s):	dnaseq@leibniz-fli.de
Organization name:	Leibniz Institute on Aging - FLI
Department:	Core Facility - DNA Sequencing
Street address:	Beutenbergstraße 11
City:	Jena
ZIP/Postal code:	07747
Country:	Germany
```

### d. How was the RNA extracted?

Strand-specific RNA-seq were prepared using the TruSeq Stranded RNA LT Kit (Illumina), and 

### e. What library prep was used?

Library preparation was done using Illumina’s TruSeq RNA Library Prep Kit v2 kit (single library layout, cDNA library selection)

### f. What cell type was used?

There are 10 cell types for each species (Guinea pig/Naked mole rat): heart—Hrt (NMR only), skin—Skn, liver—Lvr, kidney—Kid, cerebellum—Cer, hypothalamus—Hyp, pituitary—Pit, thyroid—Thy, adrenal—Adr, and gonads—Gon (testis—Tes /ovaries—Ova). I am only going to investigate either skin or liver (depending on if the Damara Mole Rat dataset if ok to use)

### g. what was the treatment/experimental condition?

The conditions of the Naked mole rat (NMR) are described in the paper:

"NMR colonies were kept inside a climatized box (2×1×1 m) in artificial burrow systems, consisting of eight cylindrical acrylic glass containers (diameter 240 mm, height 285 or 205 mm). The latter functioned as variable nest boxes, food chambers, or toilet areas, and were interconnected with acrylic tubes having an inner diameter of 60 mm. Husbandry conditions were stabile during the entire experimental period of 22 months. Temperature and humidity were adjusted to 27.0 ± 2.0 °C and 85.0 ± 5.0%, respectively. In general, the NMR colonies were kept in darkness except for 2 to 4 h of daily husbandry activities. Fresh vegetable food was provided daily and ad libitum. In addition, commercial rat pellets (Vita special, Vitakraft GmbH, Bremen, Germany) were fed as an additional source of protein and trace elements.

To turn them into breeders, randomly selected non-breeding animals derived from two long-term (> 4 years) established colonies of more than 50 individuals were separated and paired with the opposite sex. As non-breeder controls, litter siblings of paired animals remained in their colonies as workers. After the lactation period of the second set of live offspring, the tissue sampling was scheduled. To avoid further pregnancies in the females, male partners were removed and euthanized 8–10 days postpartum. The tissue collection in the females took place 40–50 days after the end of last pregnancy."

### h. What sequencing platform was used?

Illumina, instrument - Illumina HiSeq 2500

# Question 2: Starting the alignment

>Align the FASTQ file with an appropriate aligner (you may have to build a new index) Document: (3pt)
 - parameters (and why you chose them)
 - summary of outcome and basic QC
 
The reference genomes for naked mole rats are found in Ensembl. The female and male have separate reference genomes.

For Female: http://ftp.ensembl.org/pub/current_gtf/heterocephalus_glaber_female/

For Male: http://ftp.ensembl.org/pub/current_gtf/heterocephalus_glaber_male/

### a. Preparing the files

Copy the files (reference and fastq files to my directory at /athena/angsd/scratch/tof4003/project)

For now, I will copy all the accession for male, non-breeder, skin RNAs + the male reference. Then use STAR to build a reference genome and align
```bash
#download the metadata file (liver for all NMR) and put it into my directory
scp "C:/Users/tof4003/Desktop/WCM/W2022/CMPB 5004/Project/SraRunTable_liver.txt" tof4003@aphrodite.med.cornell.edu:/athena/angsd/scratch/tof4003/project
scp "C:/Users/tof4003/Desktop/WCM/W2022/CMPB 5004/Project/SraRunTable_skin.txt" tof4003@aphrodite.med.cornell.edu:/athena/angsd/scratch/tof4003/project
head -5 
```
#### Output
```
Run,Age,Assay Type,AvgSpotLen,Bases,BioProject,BioSample,Bytes,Center Name,Consent,DATASTORE filetype,DATASTORE provider,DATASTORE region,Experiment,GEO_Accession (exp),Instrument,LibraryLayout,LibrarySelection,LibrarySource,Organism,Platform,ReleaseDate,reproductive_status,Sample Name,sex,source_name,SRA Study,tissue,strain
SRR5517375,3.7 years,RNA-Seq,51,1279598517,PRJNA385839,SAMN06917115,920129275,GEO,public,"fastq,sra","gs,ncbi,s3","gs.US,ncbi.public,s3.us-east-1",SRX2790752,GSM2608665,Illumina HiSeq 2500,SINGLE,cDNA,TRANSCRIPTOMIC,Heterocephalus glaber,ILLUMINA,2018-06-14T00:00:00Z,breeder,GSM2608665,female,skin,SRP106647,skin,
SRR5517377,3.4 years,RNA-Seq,51,1502226369,PRJNA385839,SAMN06917062,1075163918,GEO,public,"fastq,sra","gs,ncbi,s3","gs.US,ncbi.public,s3.us-east-1",SRX2790754,GSM2608669,Illumina HiSeq 2500,SINGLE,cDNA,TRANSCRIPTOMIC,Heterocephalus glaber,ILLUMINA,2018-06-14T00:00:00Z,breeder,GSM2608669,male,skin,SRP106647,skin,
SRR5517379,2.7 years,RNA-Seq,51,1510153197,PRJNA385839,SAMN06917060,1069435086,GEO,public,"fastq,sra","gs,ncbi,s3","gs.US,ncbi.public,s3.us-east-1",SRX2790756,GSM2608672,Illumina HiSeq 2500,SINGLE,cDNA,TRANSCRIPTOMIC,Heterocephalus glaber,ILLUMINA,2018-06-14T00:00:00Z,non-breeder,GSM2608672,male,skin,SRP106647,skin,
SRR5517382,3.7 years,RNA-Seq,51,1954838670,PRJNA385839,SAMN06917056,1382548748,GEO,public,"fastq,sra","gs,ncbi,s3","gs.US,ncbi.public,s3.us-east-1",SRX2790759,GSM2608678,Illumina HiSeq 2500,SINGLE,cDNA,TRANSCRIPTOMIC,Heterocephalus glaber,ILLUMINA,2018-06-14T00:00:00Z,non-breeder,GSM2608678,female,skin,SRP106647,skin,
```
The run is field 1, reproduction status is field 28, sex is field 30

```bash
#download data
awk -F "," '{ if($28 == "non-breeder" && $30 == "male") print $1}' SraRunTable_skin.txt > skin_male_nbreed.txt
mkdir -p skn_male_nbreed
#had to sbatch because this process took a long time
nano fq_acc.sh
#! /bin/bash -l
#SBATCH --partition=angsd_class
#SBATCH --nodes=1
#SBATCH --ntasks=1
#SBATCH --job-name=SRA_run
#SBATCH --time=12:00:00 #HH/MM/SS
#SBATCH --mem=8G
#SBATCH --nodelist=farina.pbtech #specify the node
#load SRA toolkit
spack load sra-toolkit@2.8.2-1%gcc@6.3.0
cd /athena/angsd/scratch/tof4003/project
while read line; do
    prefetch $line
    #change to fastq format
    fastq-dump /home/tof4003/ncbi/public/sra/$line.sra -O /athena/angsd/scratch/tof4003/project/skn_male_nbreed
done < /athena/angsd/scratch/tof4003/project/skin_male_nbreed.txt
sbatch fq_acc.sh
#cleaning the temp files
rm -r /home/tof4003/ncbi
#download reference
mkdir -p NMR_ref
wget -O NMR_ref/maleNMR.gtf.gz http://ftp.ensembl.org/pub/current_gtf/heterocephalus_glaber_male/Heterocephalus_glaber_male.HetGla_1.0.105.gtf.gz
#need to also download the reference fasta dna file
wget -O NMR_ref/NMR_male_dna.fa.gz http://ftp.ensembl.org/pub/current_fasta/heterocephalus_glaber_male/dna/Heterocephalus_glaber_male.HetGla_1.0.dna.toplevel.fa.gz
#unzip everything (for some reason directly specifying zcat to read .gz file with star returns that I have an extra space before the > in fasta, even though checking using head indicates no problem)
zcat /athena/angsd/scratch/tof4003/project/NMR_ref/maleNMR.gtf.gz > /athena/angsd/scratch/ tof4003/project/NMR_ref/maleNMR.gtf
zcat /athena/angsd/scratch/tof4003/project/NMR_ref/NMR_male_dna.fa.gz > /athena/angsd/scratch/tof4003/project/NMR_ref/NMR_ref_male.fa
ls skn_male_nbreed
head -10 skn_male_nbreed/SRR5517391.fastq
ls NMR_ref
```
#### Output

ls skn_male_nbreed
```
SRR5517374.fastq SRR5517384.fastq SRR5517391.fastq
SRR5517379.fastq SRR5517389.fastq SRR5517392.fastq
```
head -10 skn_male_nbreed/SRR5517391.fastq
```
@SRR5517391.1 HISEQ2500:357:C5D1NACXX:5:1101:1459:2059 length=51
NGACTACGAGGAACTTTGCTTCGACTCTTACATAGTACCAAGCACGGATTT
+SRR5517391.1 HISEQ2500:357:C5D1NACXX:5:1101:1459:2059 length=51
#1=DDFFFGHHGHIHIHIJJIIIIJGHHI9DFFGHBFIGIJJGIIJIJJII
@SRR5517391.2 HISEQ2500:357:C5D1NACXX:5:1101:1385:2073 length=51
NCCACAGCGGCCGAGCTGCTGAAGCACCCGTTCCTGGCCAAGGCGGGGCCG
+SRR5517391.2 HISEQ2500:357:C5D1NACXX:5:1101:1385:2073 length=51
#4=DDFADCFHHDHHHGIEHIJJFHIIJE?@HHIJCHGIICGHHHC@BBDB
@SRR5517391.3 HISEQ2500:357:C5D1NACXX:5:1101:1344:2119 length=51
ATTTATTCTGGAGCAAGCAATTACAGGGGATTTTGCTTTGGTGAAAGCCTG
```
ls NMR_ref
```
maleNMR.gtf  maleNMR.gtf.gz  NMR_male_dna.fa.gz  NMR_ref_male.fa
```

### b. Creating STAR Ref Index

Now, run STAR to create genome index for STAR
```bash
#the reference is similar to the Gierlinski data set so I just used the same parameters (sjdboverhang will prevent any splice pverhang smaller than 99)
#will increase the thread and gpu to run faster
mkdir -p star_test
nano star_test/ref_gen.sh

#! /bin/bash -l
#SBATCH --partition=angsd_class
#SBATCH --nodes=1
#SBATCH --ntasks=4
#SBATCH #SBATCH --cpus-per-task=8 #try multiple threads with multiple cpu
#SBATCH --job-name=REf_gen_STAR
#SBATCH --time=48:00:00 #HH/MM/SS
#SBATCH --mem=100G
#SBATCH --nodelist=buddy.pbtech #specify the node
cd /athena/angsd/scratch/tof4003/project/NMR_ref
mkdir -p STAR_ref
#load STAR
spack load star@2.7.0e
STAR --runMode genomeGenerate \
--runThreadN 2 \
--genomeDir STAR_ref \
--genomeFastaFiles NMR_ref_male.fa \
--sjdbGTFfile maleNMR.gtf \
--limitGenomeGenerateRAM 100326568150 \
--sjdbOverhang 99

sbatch star_test/ref_gen.sh
ls NMR_ref/STAR_ref
```

#### Output
```
chrLength.txt      exonInfo.tab          SAindex
chrNameLength.txt  geneInfo.tab          sjdbInfo.txt
chrName.txt        Genome                sjdbList.fromGTF.out.tab
chrStart.txt       genomeParameters.txt  sjdbList.out.tab
exonGeTrInfo.tab   SA                    transcriptInfo.tab

```


### c. Check fastqc quality for the sequences (Optional)
```bash
nano skn_male_QC.sh
#! /bin/bash -l
#SBATCH --partition=angsd_class
#SBATCH --nodes=1
#SBATCH --ntasks=1
#SBATCH --job-name=fastqc
#SBATCH --time=12:00:00 #HH/MM/SS
#SBATCH --mem=8G
#SBATCH --nodelist=farina.pbtech #specify the node

#going to try trimming 1 file and see if there is any difference...
spack load -r trimgalore
spack load fastqc
cd /athena/angsd/scratch/tof4003/project/skn_male_nbreed
for file in *.fastq ; do
  echo $file
  fastqc $file
  if [ $file == "SRR5517389.fastq" ]; then
    echo "Trimming $file ... "
    trim_galore --illumina $file
    fv=$(cut -d "." -f 1 <<< "$file")
    sbstring="_trimmed.fq"
    newf=$(printf "$fv$sbstring")
    fastqc $newf
  fi
done
#then, run a multiqc
spack load -r py-multiqc
multiqc .

sbatch skn_male_QC.sh

ls skn_male_nbreed

#file download in cmd
scp tof4003@aphrodite.med.cornell.edu:/athena/angsd/scratch/tof4003/project/skn_male_nbreed/*.html "Downloads"
```

#### Output
```
multiqc_data            SRR5517389_fastqc.html
multiqc_report.html     SRR5517389_fastqc.zip
SRR5517374.fastq        SRR5517389.fastq_trimming_report.txt
SRR5517374_fastqc.html  SRR5517389_trimmed_fastqc.html
SRR5517374_fastqc.zip   SRR5517389_trimmed_fastqc.zip
SRR5517379.fastq        SRR5517389_trimmed.fq
SRR5517379_fastqc.html  SRR5517391.fastq
SRR5517379_fastqc.zip   SRR5517391_fastqc.html
SRR5517384.fastq        SRR5517391_fastqc.zip
SRR5517384_fastqc.html  SRR5517392.fastq
SRR5517384_fastqc.zip   SRR5517392_fastqc.html
SRR5517389.fastq        SRR5517392_fastqc.zip
```
I am including the multiQC report to this assignment. The sequences are in good quality.

#### MultiQC Sequence QC Scores

![QC Scores](\Users\tof4003\Desktop\WCM\W2022\CMPB 5004\Project\skn_male_breed_per_sequence_quality_scores_plot.png)

There is also no much difference in GC content when trimmed. So I decide that the sample does not to be trimmed

#### Untrimmed GC Content for SRR5517389

![GC Content Untrimmed](\Users\tof4003\Desktop\WCM\W2022\CMPB 5004\Project\untrimmed GC 5517389.png)

#### Trimmed GC Content for SRR5517389

![GC Content Trimmed](\Users\tof4003\Desktop\WCM\W2022\CMPB 5004\Project\trimmed GC 5517389.png)

### d. Running Aligners

### d1. BWA
```bash
mkdir -p bwa_test
cd bwa_test
nano bwa_script.sh
#! /bin/bash -l
#SBATCH --partition=angsd_class
#SBATCH --nodes=1
#SBATCH --ntasks=4
#SBATCH --cpus-per-task=4 #try multiple threads with multiple cpu
#SBATCH --job-name=BWA_tof4003
#SBATCH --time=48:00:00 #HH/MM/SS
#SBATCH --mem=12G
#SBATCH --nodelist=farina.pbtech #specify the node

#load modules
spack load bwa@0.7.15% gcc@6.3.0
spack load samtools@1.9% gcc@6.3.0
cd /athena/angsd/scratch/tof4003/project/NMR_ref
mkdir -p bwa_index_NMR_male
cd bwa_index_NMR_male
#Index the genome
bwa index -p NMR_BWA_index /athena/angsd/scratch/tof4003/project/NMR_ref/NMR_ref                                                                             _male.fa
cd /athena/angsd/scratch/tof4003/project/skn_male_nbreed
#run BWA mem: ref genome > output
mkdir -p /athena/angsd/scratch/tof4003/project/bwa_test/output
for file in *.fastq; do
  bwa mem -t 2 /athena/angsd/scratch/tof4003/project/NMR_ref/bwa_index_NMR_male/                                                                             NMR_BWA_index $file > /athena/angsd/scratch/tof4003/project/bwa_test/output/$fil                                                                             e.bwa.sam
  #use samtools to covert to bam, sort& index
  #sort & covert to bam
  samtools view -b /athena/angsd/scratch/tof4003/project/bwa_test/output/$file.b                                                                             wa.sam | samtools sort -o /athena/angsd/scratch/tof4003/project/bwa_test/output/                                                                             $file.bwa.sorted.bam
  #index
  samtools index /athena/angsd/scratch/tof4003/project/bwa_test/output/$file.bwa                                                                             .sorted.bam
  echo "bwa for $file done"
done

sbatch bwa_script.sh
```

### d2. STAR

```bash
cd /athena/angsd/scratch/tof4003/project/star_test
nano star_script.sh
#! /bin/bash -l
#SBATCH --partition=angsd_class
#SBATCH --nodes=1
#SBATCH --ntasks=4
#SBATCH --cpus-per-task=4
#SBATCH --job-name=STAR_align
#SBATCH --time=40:00:00 #HH/MM/SS
#SBATCH --mem=40G
#SBATCH --nodelist=buddy.pbtech #specify the node

#load modules
spack load star@2.7.0e
spack load samtools@1.9% gcc@6.3.0
cd /athena/angsd/scratch/tof4003/project/star_test
#running STAR
for file in /athena/angsd/scratch/tof4003/project/skn_male_nbreed/*.fastq; do
  echo "Begin $file"
  fv=$(cut -d "/" -f 8 <<< "$file" | cut -d "." -f 1 )
  STAR --runMode alignReads \
  --runThreadN 2 \
  --genomeDir /athena/angsd/scratch/tof4003/project/NMR_ref/STAR_ref \
  --readFilesIn $file \
  --outFileNamePrefix $fv.star. \
  --outSAMtype BAM SortedByCoordinate
  #use samtools to index
  #index
  samtools index $fv.star.Aligned.sortedByCoord.out.bam
  echo "star $fv done"
done

sbatch star_script.sh
```

### d3. HISAT2

Because STAR takes really long time to index, I thought to use HISAT2 also to do index and aligning,
checking the quality as well
```bash
mkdir -p test_hisat2
cd test_hisat2
nano hisat2_script.sh
#! /bin/bash -l
#SBATCH --partition=angsd_class
#SBATCH --nodes=1
#SBATCH --ntasks=4
#SBATCH --cpus-per-task=4 #try multiple threads with multiple cpu
#SBATCH --job-name=Hisat2_tof4003
#SBATCH --time=48:00:00 #HH/MM/SS
#SBATCH --mem=40G
#SBATCH --nodelist=buddy.pbtech #specify the node

#load modules
spack load hisat2
spack load samtools@1.9% gcc@6.3.0
cd /athena/angsd/scratch/tof4003/project/NMR_ref
mkdir -p hisat2_index_NMR_male
cd hisat2_index_NMR_male
#Index the genome
#hisat2-build -p 16 /athena/angsd/scratch/tof4003/project/NMR_ref/NMR_ref_male.f                                                                             a HISAT2_NMR_male
cd /athena/angsd/scratch/tof4003/project/skn_male_nbreed
#run hisat2 alignment: reads are in fastq files, threads = 2; files are not pair                                                                             -end reads
mkdir -p /athena/angsd/scratch/tof4003/project/test_hisat2/output
for file in *.fastq; do
  echo "Begin $file"
  fv=$(cut -d "." -f 1 <<< "$file")
  hisat2 -q -p 2 -x /athena/angsd/scratch/tof4003/project/NMR_ref/hisat2_index_N                                                                             MR_male/HISAT2_NMR_male -U $file > /athena/angsd/scratch/tof4003/project/test_hi                                                                             sat2/output/$fv.hisat2.sam
  #use samtools to covert to bam, sort& index
  #sort & covert to bam
  samtools view -b /athena/angsd/scratch/tof4003/project/test_hisat2/output/$fv.                                                                             hisat2.sam | samtools sort -o /athena/angsd/scratch/tof4003/project/test_hisat2/                                                                             output/$fv.hisat2.sorted.bam
  #index
  samtools index /athena/angsd/scratch/tof4003/project/test_hisat2/output/$fv.hi                                                                             sat2.sorted.bam
  echo "Hisat2 for $fv done"
done

sbatch hisat2_script.sh
```


### e. Checking Output & Quality
```bash
ls star_test 
ls bwa_test/output
spack load samtools@1.9% gcc@6.3.0
samtools view bwa_test/output/SRR5517392.fastq.bwa.sorted.bam | head -5
#hisat2 gives stats on overall alignment
cat test_hisat2/slurm-8877715.out | egrep "alignment | SRR.+"
#running bamqc
nano bamqc_test/bamqc_test.sh
#! /bin/bash -l
#SBATCH --partition=angsd_class
#SBATCH --nodes=1
#SBATCH --ntasks=1
#SBATCH --job-name=test_bamqc
#SBATCH --time=12:00:00 #HH/MM/SS
#SBATCH --mem=12G
#SBATCH --nodelist=farina.pbtech #specify the node
cd /athena/angsd/scratch/tof4003/project
/softlib/apps/EL7/BamQC/bin/bamqc -o bamqc_test bwa_test/output/*.bwa.sorted.bam star_test/*.star.Aligned.sortedByCoord.out.bam test_hisat2/output/*.hisat2.sorted.bam

cd bamqc_test
sbatch bamqc_test.sh
ls /athena/angsd/scratch/tof4003/project/bamqc_test

#file export
#Just doing 1 export file as an example to evaluate the results
scp tof4003@aphrodite.med.cornell.edu:/athena/angsd/scratch/tof4003/project/bamqc_test/SRR5517389.*.html "C:\Users\tof4003\Desktop\WCM\W2022\CMPB 5004\Project"
```

#### Output
star_test
```
ref_gen.sh
slurm-8877736.out
slurm-8879598.out
SRR5517374.star.Aligned.sortedByCoord.out.bam
SRR5517374.star.Aligned.sortedByCoord.out.bam.bai
SRR5517374.star.Log.final.out
SRR5517374.star.Log.out
SRR5517374.star.Log.progress.out
SRR5517374.star.SJ.out.tab
SRR5517379.star.Aligned.sortedByCoord.out.bam
SRR5517379.star.Aligned.sortedByCoord.out.bam.bai
SRR5517379.star.Log.final.out
SRR5517379.star.Log.out
SRR5517379.star.Log.progress.out
SRR5517379.star.SJ.out.tab
SRR5517384.star.Aligned.sortedByCoord.out.bam
SRR5517384.star.Aligned.sortedByCoord.out.bam.bai
SRR5517384.star.Log.final.out
SRR5517384.star.Log.out
SRR5517384.star.Log.progress.out
SRR5517384.star.SJ.out.tab
SRR5517389.star.Aligned.sortedByCoord.out.bam
SRR5517389.star.Aligned.sortedByCoord.out.bam.bai
SRR5517389.star.Log.final.out
SRR5517389.star.Log.out
SRR5517389.star.Log.progress.out
SRR5517389.star.SJ.out.tab
SRR5517391.star.Aligned.sortedByCoord.out.bam
SRR5517391.star.Aligned.sortedByCoord.out.bam.bai
SRR5517391.star.Log.final.out
SRR5517391.star.Log.out
SRR5517391.star.Log.progress.out
SRR5517391.star.SJ.out.tab
SRR5517392.star.Aligned.sortedByCoord.out.bam
SRR5517392.star.Aligned.sortedByCoord.out.bam.bai
SRR5517392.star.Log.final.out
SRR5517392.star.Log.out
SRR5517392.star.Log.progress.out
SRR5517392.star.SJ.out.tab
star_script.sh

```

bwa_test/output
```
SRR5517374.fastq.bwa.sam             SRR5517389.fastq.bwa.sam
SRR5517374.fastq.bwa.sorted.bam      SRR5517389.fastq.bwa.sorted.bam
SRR5517374.fastq.bwa.sorted.bam.bai  SRR5517389.fastq.bwa.sorted.bam.bai
SRR5517379.fastq.bwa.sam             SRR5517391.fastq.bwa.sam
SRR5517379.fastq.bwa.sorted.bam      SRR5517391.fastq.bwa.sorted.bam
SRR5517379.fastq.bwa.sorted.bam.bai  SRR5517391.fastq.bwa.sorted.bam.bai
SRR5517384.fastq.bwa.sam             SRR5517392.fastq.bwa.sam
SRR5517384.fastq.bwa.sorted.bam      SRR5517392.fastq.bwa.sorted.bam
SRR5517384.fastq.bwa.sorted.bam.bai  SRR5517392.fastq.bwa.sorted.bam.bai

```
head -5 for sorted SRR5517392.bwa
```
SRR5517392.14859835     0       JH173571.1      1386    46      51M     *      00       AACTCAGTTCCACATATCCTCAAGTTGCTTTCCACACCAAATTCTCTTCTT     ???;BADADD4<?BAE<CFEFEE:A?E:A9C4<CFEC@DDDEDC<9?DDDD     NM:i:0  MD:Z:51 AS:i:51 XS:i:38
SRR5517392.25704988     16      JH173571.1      1420    0       51M     *      00       CATCAAATTCTCTTCTTTCCAGTGTTATGATTTCAGGATCATCATCATAAA     HHGIGIHAIHGJHGIIJJJJIHEJIJIJJJJJIJJJJIHHHFHDFFFFCC@     NM:i:1  MD:Z:2C48       AS:i:48 XS:i:48XA:Z:JH170976.1,-615628,3S48M,0;JH172091.1,-1370896,51M,1;JH170135.1,-17093,51M,3;
SRR5517392.8349513      16      JH173571.1      1431    1       51M     *      00       CTTCTTTCCAGTGTTATGATTTCAGGATCATCATCATAAATGCCAAAATCA     ######C3@D<F??9@*9BH<A3?<,:,CA:AEA<AAA?A<<)4DBB:8;?     NM:i:1  MD:Z:41A9       AS:i:46 XS:i:44XA:Z:JH170976.1,-615636,44M7S,0;JH172091.1,-1370907,51M,2;JH170135.1,-17104,38M13S,0;
SRR5517392.26297665     16      JH173571.1      1434    0       51M     *      00       CTTTCCAGTGTTATGATTTCAGGATCATCATCATAAATGCCAAAATCATAA     D>HCBGEGAFEHFEF>GGIIIHEE?H>IIIIIEE?D@GHDFFHDD8DD@@@     NM:i:1  MD:Z:38A12      AS:i:46 XS:i:45XA:Z:JH172091.1,-1370910,51M,2;JH170976.1,-615639,41M10S,0;
SRR5517392.11952429     0       JH173571.1      1443    13      51M     *      00       GTTATGATTTCAGGATCATCATCATAAATGCCAAAATCATAACGATAATAG     CBCFFFFFHHHHHJJJFHIIJJJJJJJJJIJJJJJJJJJJIJJJJJJJIJJ     NM:i:1  MD:Z:29A21      AS:i:46 XS:i:41XA:Z:JH172091.1,+1370919,51M,2;

```
Alignment Quality for Hisat2
```
Begin SRR5517374.fastq
92.45% overall alignment rate
Hisat2 for SRR5517374 done
Begin SRR5517379.fastq
92.79% overall alignment rate
Hisat2 for SRR5517379 done
Begin SRR5517384.fastq
92.66% overall alignment rate
Hisat2 for SRR5517384 done
Begin SRR5517389.fastq
91.90% overall alignment rate
Hisat2 for SRR5517389 done
Begin SRR5517391.fastq
92.31% overall alignment rate
Hisat2 for SRR5517391 done
Begin SRR5517392.fastq
93.13% overall alignment rate
Hisat2 for SRR5517392 done

```
### Graphs

I will mainly look at the Indel frequencies for all of the outcomes. Supposedly, there are alternative splicing but the Insertions and deletions of the aligned files should have a close to constant distribution across the genome. I am attaching the reports to the homework as well, please let me know if you agree otherwise.

#### STAR Indel Frequencies
![STAR](C:\Users\tof4003\Desktop\WCM\W2022\CMPB 5004\Project\starindel.png)

#### Hisat2 Indel Frequencies
![Hisat2](C:\Users\tof4003\Desktop\WCM\W2022\CMPB 5004\Project\hisat2indel.png)

#### BWA Indel Frequencies
![BWA](C:\Users\tof4003\Desktop\WCM\W2022\CMPB 5004\Project\bwaindel.png)
I decide that the BWA gives the worst performance of all, and hisat2 gives the best performance, based on the Indel frequencies of this particular file.

# Conclusion

My evaluation for the sequences and the aligning determines that I will not need to trim the sequence before processing, and I will use Hisat2 for my seq alignment (STAR takes overwhelming amount of RAM and memory to index; however, doesn't give the best result). 
