---
title: "Project Vignette"
author: "Phoebe Fei"
date: "3/26/2022"
output: 
  html_document:
    toc: true
    toc_float: true
    toc_depth: 3
---

```{css style_settings, echo = FALSE}
blockquote { 
    font-size: 14px; 
    font-style: italic;
}
h1{
    font-size: 22px;
}
h2{
    font-size: 20px;
}
h3{
    font-size: 18px;
}
h4{
    font-size: 16px;
}

```

**This is the main workflow .rmd that guides through the workflow of this project, some copied from the checkpoint rmd**

# 1. File Download and Preparation

### a. File Download 

The reference genomes for naked mole rats are found in Ensembl. The female and male have separate reference genomes.

For Female: http://ftp.ensembl.org/pub/current_gtf/heterocephalus_glaber_female/

For Male: http://ftp.ensembl.org/pub/current_gtf/heterocephalus_glaber_male/

The data is accessible from NCBI SRA/GEO dataset under project number PRJNA386015. The link to the SRA run selector is https://www.ncbi.nlm.nih.gov/Traces/study/?acc=PRJNA386015&o=acc_s%3Aa.  

i. Copy the files (reference and fastq files to my directory at /athena/angsd/scratch/tof4003/project)

I will copy all NMR liver files to the project

```bash
#download the metadata file (liver for all NMR) and put it into my directory
scp "C:/Users/tof4003/Desktop/WCM/W2022/CMPB 5004/Project/SraRunTable_liver.txt" tof4003@aphrodite.med.cornell.edu:/athena/angsd/scratch/tof4003/project

head -5 SraRunTable_liver.txt
```
#### Output
```
Run,Age,Assay Type,AvgSpotLen,Bases,BioProject,BioSample,Bytes,Center Name,Consent,DATASTORE filetype,DATASTORE provider,DATASTORE region,Experiment,GEO_Accession (exp),Instrument,LibraryLayout,LibrarySelection,LibrarySource,Organism,Platform,ReleaseDate,reproductive_status,Sample Name,sex,source_name,SRA Study,tissue,strain
SRR5517432,3.7 years,RNA-Seq,50,1143144100,PRJNA385850,SAMN06917366,753974373,GEO,public,"fastq,sra","gs,ncbi,s3","gs.US,ncbi.public,s3.us-east-1",SRX2790801,GSM2609133,Illumina HiSeq 2500,SINGLE,cDNA,TRANSCRIPTOMIC,Heterocephalus glaber,ILLUMINA,2018-06-14T00:00:00Z,breeder,GSM2609133,female,liver,SRP106650,liver,
SRR5517435,3.6 years,RNA-Seq,50,1263344450,PRJNA385850,SAMN06917363,850450852,GEO,public,"fastq,sra","gs,ncbi,s3","gs.US,ncbi.public,s3.us-east-1",SRX2790804,GSM2609136,Illumina HiSeq 2500,SINGLE,cDNA,TRANSCRIPTOMIC,Heterocephalus glaber,ILLUMINA,2018-06-14T00:00:00Z,breeder,GSM2609136,female,liver,SRP106650,liver,
SRR5517437,3.7 years,RNA-Seq,50,1109269000,PRJNA385850,SAMN06917361,731997749,GEO,public,"fastq,sra","gs,ncbi,s3","gs.US,ncbi.public,s3.us-east-1",SRX2790806,GSM2609138,Illumina HiSeq 2500,SINGLE,cDNA,TRANSCRIPTOMIC,Heterocephalus glaber,ILLUMINA,2018-06-14T00:00:00Z,breeder,GSM2609138,female,liver,SRP106650,liver,
SRR5517441,2.4 years,RNA-Seq,51,1253621004,PRJNA385850,SAMN06917357,846076900,GEO,public,"fastq,sra","gs,ncbi,s3","gs.US,ncbi.public,s3.us-east-1",SRX2790810,GSM2609142,Illumina HiSeq 2500,SINGLE,cDNA,TRANSCRIPTOMIC,Heterocephalus glaber,ILLUMINA,2018-06-14T00:00:00Z,non-breeder,GSM2609142,female,liver,SRP106650,liver,

```
The run is field 1, reproduction status is field 28, sex is field 30


```bash

#download data
awk -F "," '{ if($30 == "male") print $1","$28}' SraRunTable_liver.txt > liver_male.txt
awk -F "," '{ if($30 == "female") print $1","$28}' SraRunTable_liver.txt > liver_female.txt

mkdir -p lvr_male
mkdir -p lvr_female

#had to sbatch because this process took a long time
nano data_acc.sh
#! /bin/bash -l
#SBATCH --partition=angsd_class
#SBATCH --account=angsd_class
#SBATCH --nodes=1
#SBATCH --ntasks=1
#SBATCH --job-name=SRA_run
#SBATCH --time=15:00:00 #HH/MM/SS
#SBATCH --mem=10G

#load SRA toolkit
spack load sra-toolkit@2.8.2-1%gcc@6.3.0
cd /athena/angsd/scratch/tof4003/project
#clear the sra directory
rm -r /home/tof4003/ncbi
while read line; do
    echo $line
    acc=$(cut -d "," -f 1 <<< "$line")
    breed=$(cut -d "," -f 2 <<< "$line")
    prefetch $acc
    #change to fastq format
    fastq-dump /home/tof4003/ncbi/public/sra/$acc.sra -O /athena/angsd/scratch/tof4003/project/lvr_male
    #rename the file
    mv /athena/angsd/scratch/tof4003/project/lvr_male/$acc.fastq /athena/angsd/scratch/tof4003/project/lvr_male/$acc.m_$breed.fastq
done < /athena/angsd/scratch/tof4003/project/liver_male.txt
rm -r /home/tof4003/ncbi
while read line; do
    echo $line
    acc=$(cut -d "," -f 1 <<< "$line")
    breed=$(cut -d "," -f 2 <<< "$line")
    prefetch $acc
    #change to fastq format
    fastq-dump /home/tof4003/ncbi/public/sra/$acc.sra -O /athena/angsd/scratch/tof4003/project/lvr_female
    #rename the file
    mv /athena/angsd/scratch/tof4003/project/lvr_female/$acc.fastq /athena/angsd/scratch/tof4003/project/lvr_female/$acc.f_$breed.fastq
done < /athena/angsd/scratch/tof4003/project/liver_female.txt

sbatch data_acc.sh
#cleaning the temp files
rm -r /home/tof4003/ncbi

#download reference
mkdir -p NMR_ref

nano ref_d.sh

#! /bin/bash -l
#SBATCH --partition=angsd_class
#SBATCH --account=angsd_class
#SBATCH --nodes=1
#SBATCH --ntasks=1
#SBATCH --job-name=SRA_run
#SBATCH --time=15:00:00 #HH/MM/SS
#SBATCH --mem=10G

cd  /athena/angsd/scratch/tof4003/project
wget -O NMR_ref/maleNMR.gtf.gz http://ftp.ensembl.org/pub/current_gtf/heterocephalus_glaber_male/Heterocephalus_glaber_male.HetGla_1.0.105.gtf.gz
wget -O NMR_ref/femaleNMR.gtf.gz http://ftp.ensembl.org/pub/current_gtf/heterocephalus_glaber_female/Heterocephalus_glaber_female.HetGla_female_1.0.105.gtf.gz
#need to also download the reference fasta dna file
wget -O NMR_ref/NMR_male_dna.fa.gz http://ftp.ensembl.org/pub/current_fasta/heterocephalus_glaber_male/dna/Heterocephalus_glaber_male.HetGla_1.0.dna.toplevel.fa.gz
wget -O NMR_ref/NMR_female_dna.fa.gz http://ftp.ensembl.org/pub/current_fasta/heterocephalus_glaber_female/dna/Heterocephalus_glaber_female.HetGla_female_1.0.dna.toplevel.fa.gz


#unzip everything (for some reason directly specifying zcat to read .gz file with star returns that I have an extra space before the > in fasta, even though checking using head indicates no problem)
zcat /athena/angsd/scratch/tof4003/project/NMR_ref/maleNMR.gtf.gz > /athena/angsd/scratch/tof4003/project/NMR_ref/maleNMR.gtf
zcat /athena/angsd/scratch/tof4003/project/NMR_ref/NMR_male_dna.fa.gz > /athena/angsd/scratch/tof4003/project/NMR_ref/NMR_ref_male.fa
zcat /athena/angsd/scratch/tof4003/project/NMR_ref/femaleNMR.gtf.gz > /athena/angsd/scratch/tof4003/project/NMR_ref/femaleNMR.gtf
zcat /athena/angsd/scratch/tof4003/project/NMR_ref/NMR_female_dna.fa.gz > /athena/angsd/scratch/tof4003/project/NMR_ref/NMR_ref_female.fa

sbatch ref_d.sh

ls lvr_female
ls lvr_male
ls NMR_ref
```
#### Output

lvr_female

```
SRR5517432.f_breeder.fastq  SRR5517441.f_non-breeder.fastq
SRR5517435.f_breeder.fastq  SRR5517445.f_non-breeder.fastq
SRR5517436.f_breeder.fastq  SRR5517446.f_non-breeder.fastq
SRR5517437.f_breeder.fastq  SRR5517449.f_non-breeder.fastq
SRR5517439.f_breeder.fastq  SRR5517450.f_non-breeder.fastq
SRR5517440.f_breeder.fastq

```
lvr_male
```
SRR5517433.m_breeder.fastq      SRR5517447.m_breeder.fastq
SRR5517434.m_breeder.fastq      SRR5517448.m_non-breeder.fastq
SRR5517438.m_breeder.fastq      SRR5517451.m_non-breeder.fastq
SRR5517442.m_non-breeder.fastq  SRR5517452.m_non-breeder.fastq
SRR5517443.m_breeder.fastq      SRR5517453.m_non-breeder.fastq
SRR5517454.m_breeder.fastq
SRR5517444.m_non-breeder.fastq

```

ls NMR_ref
STAR, bwa and hisat2 for male index were already generated and put into the folder
```
bwa_index_NMR_male     Log.out         NMR_female_dna.fa.gz  NMR_ref_male.fa
femaleNMR.gtf.gz       maleNMR.gtf     NMR_male_dna.fa.gz    STAR_ref
hisat2_index_NMR_male  maleNMR.gtf.gz  NMR_ref_female.fa

```

### b. creating Index

The STAR index for male has already been created in prev. Checkpoint using command below. (Edit: I rerun it)


```bash
#the reference is similar to the Gierlinski data set so I just used the same parameters (sjdboverhang will prevent any splice pverhang smaller than 99)
#will increase the thread and gpu to run faster
mkdir -p star_test
nano star_test/ref_gen.sh

#! /bin/bash -l
#SBATCH --partition=angsd_class
#SBATCH --account=angsd_class
#SBATCH --nodes=1
#SBATCH #SBATCH --cpus-per-task=4 #try multiple threads with multiple cpu
#SBATCH --job-name=REf_gen_STAR
#SBATCH --time=48:00:00 #HH/MM/SS
#SBATCH --mem=100G

cd /athena/angsd/scratch/tof4003/project/NMR_ref
mkdir -p STAR_ref
#load STAR
spack load star@2.7.0e
STAR --runMode genomeGenerate \
--runThreadN 4 \
--genomeDir STAR_ref \
--genomeFastaFiles NMR_ref_male.fa \
--sjdbGTFfile maleNMR.gtf \
--limitGenomeGenerateRAM 100326568150 \
--sjdbOverhang 99

sbatch star_test/ref_gen.sh
ls NMR_ref/STAR_ref
```

#### Output 

```
chrLength.txt      exonInfo.tab          SAindex
chrNameLength.txt  geneInfo.tab          sjdbInfo.txt
chrName.txt        Genome                sjdbList.fromGTF.out.tab
chrStart.txt       genomeParameters.txt  sjdbList.out.tab
exonGeTrInfo.tab   SA                    transcriptInfo.tab

```

Now, similarly I will create 1 for NMR female, with editing the resource request to speed up as suggested

```bash

mkdir -p star_liver
nano star_liver/star_female_ref.sh

#! /bin/bash -l
#SBATCH --partition=angsd_class
#SBATCH --account=angsd_class
#SBATCH --nodes=1
#SBATCH #SBATCH --cpus-per-task=4 #try multiple threads with multiple cpu
#SBATCH --job-name=REf_gen_STAR
#SBATCH --time=48:00:00 #HH/MM/SS
#SBATCH --mem=100G

cd /athena/angsd/scratch/tof4003/project/NMR_ref
mkdir -p STAR_ref_female
#load STAR
spack load star@2.7.0e
STAR --runMode genomeGenerate \
--runThreadN 4 \
--genomeDir STAR_ref_female \
--genomeFastaFiles NMR_ref_female.fa \
--sjdbGTFfile femaleNMR.gtf \
--limitGenomeGenerateRAM 100326568150 \
--sjdbOverhang 99

sbatch star_liver/star_female_ref.sh
```


### c. Checking Quality of fastqc files

A subset of the samples is done already in Checkpoint1 - see rendered html.

# 2. Alignment

I will run STAR and Hisat2 - STAR is extremely slow but since it's the standard, I will run it. Hisat2 has generally stable performance.


### a. Hisat2

```bash
mkdir -p liver_hisat2
cd liver_hisat2

nano hisat2_script.sh
#! /bin/bash -l
#SBATCH --partition=angsd_class
#SBATCH --account=angsd_class
#SBATCH --nodes=1
#SBATCH --cpus-per-task=4 #try multiple threads with multiple cpu
#SBATCH --job-name=Hisat2_tof4003
#SBATCH --time=48:00:00 #HH/MM/SS
#SBATCH --mem=60G

#load modules
spack load hisat2
spack load samtools@1.9% gcc@6.3.0
cd /athena/angsd/scratch/tof4003/project/NMR_ref
#male has already been indexed, need female
mkdir -p hisat2_index_NMR_female
cd hisat2_index_NMR_female
#Index the genome
hisat2-build -p 16 /athena/angsd/scratch/tof4003/project/NMR_ref/NMR_ref_female.fa HISAT2_NMR_female
cd /athena/angsd/scratch/tof4003/project/lvr_male
#run hisat2 alignment: reads are in fastq files, threads = 2; files are not pair-end reads
mkdir -p /athena/angsd/scratch/tof4003/project/liver_hisat2/out_male
for file in *.fastq; do
  echo "Begin $file"
  fv=$(cut -d "." -f 1 <<< "$file")
  suf=$(cut -d "." -f 2 <<< "$file")
  hisat2 -q -p 2 -x /athena/angsd/scratch/tof4003/project/NMR_ref/hisat2_index_NMR_male/HISAT2_NMR_male -U $file > /athena/angsd/scratch/tof4003/project/liver_hisat2/out_male/$fv.$suf.hisat2.sam
  #use samtools to covert to bam, sort& index
  #sort & covert to bam
  samtools view -b /athena/angsd/scratch/tof4003/project/liver_hisat2/out_male/$fv.$suf.hisat2.sam | samtools sort -o /athena/angsd/scratch/tof4003/project/liver_hisat2/out_male/$fv.$suf.hisat2.sorted.bam
  #index
  samtools index /athena/angsd/scratch/tof4003/project/liver_hisat2/out_male/$fv.$suf.hisat2.sorted.bam
  echo "Hisat2 for $fv.$suf done"
done

#females
cd /athena/angsd/scratch/tof4003/project/lvr_female

mkdir -p /athena/angsd/scratch/tof4003/project/liver_hisat2/out_female
for file in *.fastq; do
  echo "Begin $file"
  fv=$(cut -d "." -f 1 <<< "$file")
  suf=$(cut -d "." -f 2 <<< "$file")
  hisat2 -q -p 2 -x /athena/angsd/scratch/tof4003/project/NMR_ref/hisat2_index_NMR_female/HISAT2_NMR_female -U $file > /athena/angsd/scratch/tof4003/project/liver_hisat2/out_female/$fv.$suf.hisat2.sam
  #use samtools to covert to bam, sort& index
  #sort & covert to bam
  samtools view -b /athena/angsd/scratch/tof4003/project/liver_hisat2/out_female/$fv.$suf.hisat2.sam | samtools sort -o /athena/angsd/scratch/tof4003/project/liver_hisat2/out_female/$fv.$suf.hisat2.sorted.bam
  #index
  samtools index /athena/angsd/scratch/tof4003/project/liver_hisat2/out_female/$fv.$suf.hisat2.sorted.bam
  echo "Hisat2 for $fv.$suf done"
done

sbatch hisat2_script.sh

#remove .sam files to free up space
rm /athena/angsd/scratch/tof4003/project/liver_hisat2/out_female/*.sam
rm /athena/angsd/scratch/tof4003/project/liver_hisat2/out_male/*.sam
```

### b. STAR

```bash
cd /athena/angsd/scratch/tof4003/project/star_liver

nano star_align.sh
#star actually gives core-dump issue for male genomes if not enough core is reserved. 

#! /bin/bash -l
#SBATCH --partition=angsd_class
#SBATCH --account=angsd_class
#SBATCH --nodes=1
#SBATCH --cpus-per-task=8
#SBATCH --job-name=STAR_align
#SBATCH --time=40:00:00 #HH/MM/SS
#SBATCH --mem=90G

#load modules
spack load star@2.7.0e
spack load samtools@1.9% gcc@6.3.0
cd /athena/angsd/scratch/tof4003/project/star_liver
#running STAR
for file in /athena/angsd/scratch/tof4003/project/lvr_male/*.fastq; do
  echo "Begin $file"
  fv=$(cut -d "/" -f 8 <<< "$file" | cut -d "." -f 1 )
  suf=$(cut -d "/" -f 8 <<< "$file" | cut -d "." -f 2)
  STAR --runMode alignReads \
  --runThreadN 8 \
  --genomeDir /athena/angsd/scratch/tof4003/project/NMR_ref/STAR_ref \
  --readFilesIn $file \
  --outFileNamePrefix $fv.$suf.star. \
  --outSAMtype BAM SortedByCoordinate
  #use samtools to index
  #index
  samtools index $fv.$suf.star.Aligned.sortedByCoord.out.bam
  echo "star $fv.$suf done"
done
#females
for file in /athena/angsd/scratch/tof4003/project/lvr_female/*.fastq; do
  echo "Begin $file"
  fv=$(cut -d "/" -f 8 <<< "$file" | cut -d "." -f 1 )
  suf=$(cut -d "/" -f 8 <<< "$file" | cut -d "." -f 2)
  STAR --runMode alignReads \
  --runThreadN 4 \
  --genomeDir /athena/angsd/scratch/tof4003/project/NMR_ref/STAR_ref_female \
  --readFilesIn $file \
  --outFileNamePrefix $fv.$suf.star. \
  --outSAMtype BAM SortedByCoordinate
  #use samtools to index
  #index
  samtools index $fv.$suf.star.Aligned.sortedByCoord.out.bam
  echo "star $fv.$suf done"
done

sbatch star_align.sh
```

# 3.  Quality Check

### a. clearing the file and quick check in the folder

```bash
cd /athena/angsd/scratch/tof4003/project

#look at hisat2 alignment status
cat liver_hisat2/slurm-8956337.out | egrep "alignment | SRR.+"
#there is a separate file for females
cat liver_hisat2/slurm-8959476.out | egrep "alignment | SRR.+"

#freeing some memories
nano gz.sh

#! /bin/bash -l
#SBATCH --partition=angsd_class
#SBATCH --account=angsd_class
#SBATCH --nodes=1
#SBATCH --ntasks=1
#SBATCH --job-name=zip_files
#SBATCH --time=15:00:00 #HH/MM/SS
#SBATCH --mem=8G

cd /athena/angsd/scratch/tof4003/project
gzip lvr_male/*.fastq lvr_male/*.fastq.gz
rm lvr_male/*.fastq
gzip lvr_female/*.fastq lvr_female/*fastq.gz
rm lvr_female/*.fastq
rm liver_hisat2/out_female/*.sam

sbatch gz.sh

#separate out the female and female files for star
mkdir -p star_liver/female_out
mkdir -p star_liver/male_out

mv star_liver/*f_* star_liver/female_out
mv star_liver/*m_* star_liver/male_out
```

#### Output

Hisat2 male alignment (the job was killed before all female was done so I deleted the female alignments and reran, also removed below the records for females in slurm-8956337.out)
```
Begin SRR5517433.m_breeder.fastq
92.72% overall alignment rate
Hisat2 for SRR5517433.m_breeder done
Begin SRR5517434.m_breeder.fastq
93.00% overall alignment rate
Hisat2 for SRR5517434.m_breeder done
Begin SRR5517438.m_breeder.fastq
93.16% overall alignment rate
Hisat2 for SRR5517438.m_breeder done
Begin SRR5517442.m_non-breeder.fastq
94.02% overall alignment rate
Hisat2 for SRR5517442.m_non-breeder done
Begin SRR5517443.m_breeder.fastq
93.89% overall alignment rate
Hisat2 for SRR5517443.m_breeder done
Begin SRR5517444.m_non-breeder.fastq
93.69% overall alignment rate
Hisat2 for SRR5517444.m_non-breeder done
Begin SRR5517447.m_breeder.fastq
93.17% overall alignment rate
Hisat2 for SRR5517447.m_breeder done
Begin SRR5517448.m_non-breeder.fastq
92.37% overall alignment rate
Hisat2 for SRR5517448.m_non-breeder done
Begin SRR5517451.m_non-breeder.fastq
90.07% overall alignment rate
Hisat2 for SRR5517451.m_non-breeder done
Begin SRR5517452.m_non-breeder.fastq
93.72% overall alignment rate
Hisat2 for SRR5517452.m_non-breeder done
Begin SRR5517454.m_breeder.fastq
93.39% overall alignment rate
Hisat2 for SRR5517454.m_breeder done
Begin SRR5517453.m_non-breeder.fastq
93.80% overall alignment rate
Hisat2 for SRR5517454.m_non-breeder done

```

Hisat2 female alignment
```
Begin SRR5517432.f_breeder.fastq
93.37% overall alignment rate
Hisat2 for SRR5517432.f_breeder done
Begin SRR5517435.f_breeder.fastq
93.33% overall alignment rate
Hisat2 for SRR5517435.f_breeder done
Begin SRR5517436.f_breeder.fastq
92.50% overall alignment rate
Hisat2 for SRR5517436.f_breeder done
Begin SRR5517437.f_breeder.fastq
92.23% overall alignment rate
Hisat2 for SRR5517437.f_breeder done
Begin SRR5517439.f_breeder.fastq
92.76% overall alignment rate
Hisat2 for SRR5517439.f_breeder done
Begin SRR5517440.f_breeder.fastq
Hisat2 for SRR5517440.f_breeder done
Begin SRR5517441.f_non-breeder.fastq
Hisat2 for SRR5517441.f_non-breeder done
Begin SRR5517445.f_non-breeder.fastq
Hisat2 for SRR5517445.f_non-breeder done
Begin SRR5517446.f_non-breeder.fastq
Hisat2 for SRR5517446.f_non-breeder done
Begin SRR5517449.f_non-breeder.fastq
Hisat2 for SRR5517449.f_non-breeder done
```


### b. fastqc

```bash
mkdir -p fastqc_bam
cd fastqc_bam
mkdir -p star
mkdir -p hisat2

nano bam_fastqc.sh

#! /bin/bash -l
#SBATCH --partition=angsd_class
#SBATCH --account=angsd_class
#SBATCH --nodes=1
#SBATCH --job-name=bam_fastqc
#SBATCH --time=12:00:00 #HH/MM/SS
#SBATCH --mem=14G

# will access star and hisat2 separately
spack load fastqc
cd /athena/angsd/scratch/tof4003/project/liver_hisat2/out_male
for file in *.hisat2.sorted.bam ; do
  echo $file
  fastqc $file -o /athena/angsd/scratch/tof4003/project/fastqc_bam/hisat2
done
cd /athena/angsd/scratch/tof4003/project/liver_hisat2/out_female
for file in *.hisat2.sorted.bam ; do
  echo $file
  fastqc $file -o /athena/angsd/scratch/tof4003/project/fastqc_bam/hisat2
done
cd /athena/angsd/scratch/tof4003/project/star_liver/female_out
for file in *.star.Aligned.sortedByCoord.out.bam ; do
  echo $file
  fastqc $file -o /athena/angsd/scratch/tof4003/project/fastqc_bam/star
done
cd /athena/angsd/scratch/tof4003/project/star_liver/male_out
for file in *.star.Aligned.sortedByCoord.out.bam ; do
  echo $file
  fastqc $file -o /athena/angsd/scratch/tof4003/project/fastqc_bam/star
done

#then, run multiqc
cd /athena/angsd/scratch/tof4003/project/fastqc_bam/star
spack load -r py-multiqc
multiqc .
cd /athena/angsd/scratch/tof4003/project/fastqc_bam/hisat2
multiqc .

sbatch bam_fastqc.sh

#file download in cmd
scp tof4003@aphrodite.med.cornell.edu:/athena/angsd/scratch/tof4003/project/fastqc_bam/hisat2/multiqc_report.html "C:\Users\tof4003\Desktop\WCM\W2022\CMPB 5004\Project\ANGSD Project tof4003\QC htmls\hisat2_fastqc_bam.html"

scp tof4003@aphrodite.med.cornell.edu:/athena/angsd/scratch/tof4003/project/fastqc_bam/star/multiqc_report.html "C:\Users\tof4003\Desktop\WCM\W2022\CMPB 5004\Project\ANGSD Project tof4003\QC htmls\star_fastqc_bam.html"
```

#### Output of Some Selected Features



### c. bamqc

```bash
#running bamqc
mkdir -p bamqc_liver
cd bamqc_liver
nano bamqc_test.sh

#! /bin/bash -l
#SBATCH --partition=angsd_class
#SBATCH --account=angsd_class
#SBATCH --nodes=1
#SBATCH --job-name=test_bamqc
#SBATCH --time=12:00:00 #HH/MM/SS
#SBATCH --mem=12G

cd /athena/angsd/scratch/tof4003/project
/softlib/apps/EL7/BamQC/bin/bamqc -o bamqc_liver star_liver/female_out/*.star.Aligned.sortedByCoord.out.bam
star_liver/male_out/*.star.Aligned.sortedByCoord.out.bam
liver_hisat2/out_male/*.hisat2.sorted.bam
liver_hisat2/out_female/*.hisat2.sorted.bam

sbatch bamqc_test.sh

ls /athena/angsd/scratch/tof4003/project/bamqc_liver

#file transfer of a sample of files

scp tof4003@aphrodite.med.cornell.edu:/athena/angsd/scratch/tof4003/project/bamqc_liver/*.html "C:\Users\tof4003\Desktop\WCM\W2022\CMPB 5004\Project\ANGSD Project tof4003\QC htmls"


```

#### Output of Some Samples


# 4. Feature Counts

```bash

cd /athena/angsd/scratch/tof4003/project

#will generate count tables for star and hisat2 separately
mkdir -p featureCounts
cd featureCounts

nano featCounts.sh

#! /bin/bash -l
#SBATCH --partition=angsd_class
#SBATCH --account=angsd_class
#SBATCH --nodes=1
#SBATCH --job-name=feat_counts
#SBATCH --time=12:00:00 #HH/MM/SS
#SBATCH --mem=15G

# for the -O I am not going to include the "-O" reads can be assigned to more than 1 matched exon (can be counted more than once) - so if reads matched more than one exon, all of them will be counted. As discussed in the hw, the ambiguous reads are most likely UTR's so I actually don't need them
cd /athena/angsd/scratch/tof4003/project
spack load subread
# for the -O I am not going to include the "-O" reads can be assigned to more than 1 matched exon (can be counted more than once) - so if reads matched more than one exon, all of them will be counted. As discussed in the hw, the ambiguous reads are most likely UTR's so I actually don't need them
# male and female needs to be done separately (2 references)

featureCounts -a NMR_ref/maleNMR.gtf \
-o featureCounts/star_featCounts_maleNMR.txt \
-t exon \
-g gene_id \
-f \
-G NMR_ref/NMR_ref_male.fa \
star_liver/male_out/*.bam

featureCounts -a NMR_ref/femaleNMR.gtf \
-o featureCounts/star_featCounts_femaleNMR.txt \
-t exon \
-g gene_id \
-f \
-G NMR_ref/NMR_ref_female.fa \
star_liver/female_out/*.bam

featureCounts -a NMR_ref/maleNMR.gtf \
-o featureCounts/hisat2_featCounts_maleNMR.txt \
-t exon \
-g gene_id \
-f \
-G NMR_ref/NMR_ref_male.fa \
liver_hisat2/out_male/*.bam

featureCounts -a NMR_ref/femaleNMR.gtf \
-o featureCounts/hisat2_featCounts_femaleNMR.txt \
-t exon \
-g gene_id \
-f \
-G NMR_ref/NMR_ref_female.fa \
liver_hisat2/out_female/*.bam

sbatch featCounts.sh
cat *.out


--------------file download

scp tof4003@aphrodite.med.cornell.edu:/athena/angsd/scratch/tof4003/project/featureCounts/*.txt "C:\Users\tof4003\Desktop\WCM\W2022\CMPB 5004\Project\ANGSD Project tof4003"
scp tof4003@aphrodite.med.cornell.edu:/athena/angsd/scratch/tof4003/project/featureCounts/*.txt.summary "C:\Users\tof4003\Desktop\WCM\W2022\CMPB 5004\Project\ANGSD Project tof4003"

```

#### Output

showing feature counts for 1
```

        ==========     _____ _    _ ____  _____  ______          _____
        =====         / ____| |  | |  _ \|  __ \|  ____|   /\   |  __ \
          =====      | (___ | |  | | |_) | |__) | |__     /  \  | |  | |
            ====      \___ \| |  | |  _ <|  _  /|  __|   / /\ \ | |  | |
              ====    ____) | |__| | |_) | | \ \| |____ / ____ \| |__| |
        ==========   |_____/ \____/|____/|_|  \_\______/_/    \_\_____/
          v1.6.2

//========================== featureCounts setting ===========================\\
||                                                                            ||
||             Input files : 11 BAM files                                     ||
||                           S SRR5517432.f_breeder.hisat2.sorted.bam         ||
||                           S SRR5517435.f_breeder.hisat2.sorted.bam         ||
||                           S SRR5517436.f_breeder.hisat2.sorted.bam         ||
||                           S SRR5517437.f_breeder.hisat2.sorted.bam         ||
||                           S SRR5517439.f_breeder.hisat2.sorted.bam         ||
||                           S SRR5517440.f_breeder.hisat2.sorted.bam         ||
||                           S SRR5517441.f_non-breeder.hisat2.sorted.bam     ||
||                           S SRR5517445.f_non-breeder.hisat2.sorted.bam     ||
||                           S SRR5517446.f_non-breeder.hisat2.sorted.bam     ||
||                           S SRR5517449.f_non-breeder.hisat2.sorted.bam     ||
||                           S SRR5517450.f_non-breeder.hisat2.sorted.bam     ||
||                                                                            ||
||             Output file : hisat2_featCounts_femaleNMR.txt                  ||
||                 Summary : hisat2_featCounts_femaleNMR.txt.summary          ||
||              Annotation : femaleNMR.gtf (GTF)                              ||
||      Dir for temp files : featureCounts                                    ||
||                                                                            ||
||                 Threads : 1                                                ||
||                   Level : feature level                                    ||
||              Paired-end : no                                               ||
||      Multimapping reads : not counted                                      ||
|| Multi-overlapping reads : not counted                                      ||
||   Min overlapping bases : 1                                                ||
||                                                                            ||
\\===================== http://subread.sourceforge.net/ ======================//

//================================= Running ==================================\\
||                                                                            ||
|| Load annotation file femaleNMR.gtf ...                                     ||
||    Features : 328743                                                       ||
||    Meta-features : 31806                                                   ||
||    Chromosomes/contigs : 460                                               ||
||                                                                            ||
|| Loading FASTA contigs : NMR_ref/NMR_ref_female.fa                          ||
||    4229 contigs were loaded                                                ||
||                                                                            ||
|| Process BAM file SRR5517432.f_breeder.hisat2.sorted.bam...                 ||
||    Single-end reads are included.                                          ||
||    Assign reads to features...                                             ||
||    Total reads : 24624992                                                  ||
||    Successfully assigned reads : 8968581 (36.4%)                           ||
||    Running time : 0.41 minutes                                             ||
||                                                                            ||
|| Process BAM file SRR5517435.f_breeder.hisat2.sorted.bam...                 ||
||    Single-end reads are included.                                          ||
||    Assign reads to features...                                             ||
||    Total reads : 26998998                                                  ||
||    Successfully assigned reads : 9254783 (34.3%)                           ||
||    Running time : 0.45 minutes                                             ||
||                                                                            ||
|| Process BAM file SRR5517436.f_breeder.hisat2.sorted.bam...                 ||
||    Single-end reads are included.                                          ||
||    Assign reads to features...                                             ||
||    Total reads : 24683617                                                  ||
||    Successfully assigned reads : 9094694 (36.8%)                           ||
||    Running time : 0.40 minutes                                             ||
||                                                                            ||
|| Process BAM file SRR5517437.f_breeder.hisat2.sorted.bam...                 ||
||    Single-end reads are included.                                          ||
||    Assign reads to features...                                             ||
||    Total reads : 23891994                                                  ||
||    Successfully assigned reads : 8321472 (34.8%)                           ||
||    Running time : 0.41 minutes                                             ||
||                                                                            ||
|| Process BAM file SRR5517439.f_breeder.hisat2.sorted.bam...                 ||
||    Single-end reads are included.                                          ||
||    Assign reads to features...                                             ||
||    Total reads : 26804951                                                  ||
||    Successfully assigned reads : 9303414 (34.7%)                           ||
||    Running time : 0.47 minutes                                             ||
||                                                                            ||
|| Process BAM file SRR5517440.f_breeder.hisat2.sorted.bam...                 ||
||    Single-end reads are included.                                          ||
||    Assign reads to features...                                             ||
||    Total reads : 24055201                                                  ||
||    Successfully assigned reads : 9171872 (38.1%)                           ||
||    Running time : 0.40 minutes                                             ||
||                                                                            ||
|| Process BAM file SRR5517441.f_non-breeder.hisat2.sorted.bam...             ||
||    Single-end reads are included.                                          ||
||    Assign reads to features...                                             ||
||    Total reads : 26251990                                                  ||
||    Successfully assigned reads : 9936838 (37.9%)                           ||
||    Running time : 0.44 minutes                                             ||
||                                                                            ||
|| Process BAM file SRR5517445.f_non-breeder.hisat2.sorted.bam...             ||
||    Single-end reads are included.                                          ||
||    Assign reads to features...                                             ||
||    Total reads : 27278979                                                  ||
||    Successfully assigned reads : 9862489 (36.2%)                           ||
||    Running time : 0.46 minutes                                             ||
||                                                                            ||
|| Process BAM file SRR5517446.f_non-breeder.hisat2.sorted.bam...             ||
||    Single-end reads are included.                                          ||
||    Assign reads to features...                                             ||
||    Total reads : 39704440                                                  ||
||    Successfully assigned reads : 14759259 (37.2%)                          ||
||    Running time : 0.65 minutes                                             ||
||                                                                            ||
|| Process BAM file SRR5517449.f_non-breeder.hisat2.sorted.bam...             ||
||    Single-end reads are included.                                          ||
||    Assign reads to features...                                             ||
||    Total reads : 25478196                                                  ||
||    Successfully assigned reads : 9415787 (37.0%)                           ||
||    Running time : 0.42 minutes                                             ||
||                                                                            ||
|| Process BAM file SRR5517450.f_non-breeder.hisat2.sorted.bam...             ||
||    Single-end reads are included.                                          ||
||    Assign reads to features...                                             ||
||    Total reads : 30450808                                                  ||
||    Successfully assigned reads : 10357471 (34.0%)                          ||
||    Running time : 0.50 minutes                                             ||
||                                                                            ||
||                         Read assignment finished.                          ||
||                                                                            ||
|| Summary of counting results can be found in file "featureCounts/hisat2_fe  ||
|| atCounts_femaleNMR.txt.summary"                                            ||
||                                                                            ||
\\===================== http://subread.sourceforge.net/ ======================//

```

# 5. Analyze Count Data

### a. Import and process the summary Table
```{r}
library(ggplot2)
library(magrittr)
direct <- '/Users/tof4003/Desktop/WCM/W2022/CMPB 5004/Project/ANGSD Project tof4003/'
#hisat2 males & females
rchisat2f <-paste0(direct,"hisat2_featCounts_femaleNMR.txt.summary") %>%
  read.table(., header = TRUE)

rchisat2m <-paste0(direct,"hisat2_featCounts_maleNMR.txt.summary") %>%
  read.table(., header = TRUE)

#star males & females
rcstarf <-paste0(direct,"star_featCounts_femaleNMR.txt.summary") %>%
  read.table(., header = TRUE)

rcstarm <-paste0(direct,"star_featCounts_maleNMR.txt.summary") %>%
  read.table(., header = TRUE)
```
Clean the sample names

```{r}
library(stringr)
#reassign sample names
nameliststarm <- str_extract(names(rcstarm), "SRR[0-9]+.(m|f)_[a-z]+")
namelisthisat2m <- str_extract(names(rchisat2m), "SRR[0-9]+.(m|f)_[a-z]+")
nameliststarf <- str_extract(names(rcstarf), "SRR[0-9]+.(m|f)_[a-z]+")
namelisthisat2f <- str_extract(names(rchisat2f), "SRR[0-9]+.(m|f)_[a-z]+")
```
```{r}
#don't want the status to be modified
names(rcstarm)[2:length(rcstarm)] <- nameliststarm[2:length(nameliststarm)]
names(rcstarf)[2:length(rcstarf)] <- nameliststarf[2:length(nameliststarf)]
names(rchisat2m)[2:length(rchisat2m)] <- namelisthisat2m[2:length(namelisthisat2m)]
names(rchisat2f)[2:length(rchisat2f)] <- namelisthisat2f[2:length(namelisthisat2f)]
str(rchisat2f)
str(rchisat2m)
str(rcstarf)
str(rcstarm)
```

Plotting
```{r}
library(patchwork)
library(reshape2)
#STAR M
melt_starm <- melt(rcstarm, id.vars = "Status")
names(melt_starm)[3] <- "Counts"
names(melt_starm)[2] <- "Sample"
#remove 0 instances
melt_starm <- subset(melt_starm, Counts!=0)
#add a row saying identifying this data
melt_starm["Type"] <- rep("Feature Count on Male using STAR",nrow(melt_starm))
#STAR F
melt_starf <- melt(rcstarf, id.vars = "Status")
names(melt_starf)[3] <- "Counts"
names(melt_starf)[2] <- "Sample"
melt_starf <- subset(melt_starf, Counts!=0)
melt_starf["Type"] <- rep("Feature Count on Female using STAR",nrow(melt_starf))

#Hisat2 M
melt_hisat2m <- melt(rchisat2m, id.vars = "Status")
names(melt_hisat2m)[3] <- "Counts"
names(melt_hisat2m)[2] <- "Sample"
melt_hisat2m <- subset(melt_hisat2m, Counts!=0)
melt_hisat2m["Type"] <- rep("Feature Count on Male using Hisat2",nrow(melt_hisat2m))

#Hisat2 F
melt_hisat2f <- melt(rchisat2f, id.vars = "Status")
names(melt_hisat2f)[3] <- "Counts"
names(melt_hisat2f)[2] <- "Sample"
melt_hisat2f <- subset(melt_hisat2f, Counts!=0)
melt_hisat2f["Type"] <- rep("Feature Count on Male using Hisat2",nrow(melt_hisat2f))
```
Male Alignment : STAR vs. Hisat2

```{r}
melt_m <- rbind(melt_starm,melt_hisat2m)
melt_f <- rbind(melt_starf,melt_hisat2f)
```

```{r}
#size figure size
knitr::opts_chunk$set(fig.width=unit(30,"cm"), fig.height=unit(50,"cm"))
#set texts size
theme_set(theme(legend.position = "bottom", legend.key.size = unit(0.2, 'cm'),axis.text=element_text(size=8), axis.title=element_text(size=8)))
#bar plot the combined graph
plotm <- ggplot(data = melt_m,aes(x = Counts, y = Sample))+geom_bar(stat = 'identity', aes(fill = Status),position = "dodge")+facet_wrap(Type~., nrow =2, strip.position = "right")+ggtitle("Plot Based on Summary .txt Files")
plotm

```

```{r}
#size figure size
knitr::opts_chunk$set(fig.width=unit(55,"cm"), fig.height=unit(70,"cm"))
#set texts size
theme_set(theme(legend.position = "bottom", legend.key.size = unit(0.2, 'cm'),axis.text=element_text(size=8), axis.title=element_text(size=8)))
#bar plot the combined graph
plotf <- ggplot(data = melt_f,aes(x = Counts, y = Sample))+geom_bar(stat = 'identity', aes(fill = Status),width = 1,position = "dodge")+facet_wrap(Type~., nrow =2, strip.position = "right")+ggtitle("Plot Based on Summary .txt Files")
plotf


```

### b. create DESeq2 Objects

Import readcounts and assign sample names
```{r}
rchisat_f <-paste0(direct,"hisat2_featCounts_femaleNMR.txt") %>% read.table(., header = TRUE)
rchisat_m <- paste0(direct,"hisat2_featCounts_maleNMR.txt") %>% read.table(., header = TRUE)
rcstar_f <-paste0(direct,"star_featCounts_femaleNMR.txt") %>% read.table(., header = TRUE)
rcstar_m <- paste0(direct,"star_featCounts_maleNMR.txt") %>% read.table(., header = TRUE)

```
```{r}
#don't want the status to be modified
names(rcstar_m)[7:length(rcstar_m)] <- nameliststarm[2:length(nameliststarm)]
names(rcstar_f)[7:length(rcstar_f)] <- nameliststarf[2:length(nameliststarf)]
names(rchisat_m)[7:length(rchisat_m)] <- namelisthisat2m[2:length(namelisthisat2m)]
names(rchisat_f)[7:length(rchisat_f)] <- namelisthisat2f[2:length(namelisthisat2f)]

head(rcstar_m)
head(rcstar_f)
head(rchisat_m)
head(rchisat_f)
```

Creating DESeq2 Objects
```{r, eval = FALSE}
BiocManager::install("DESeq2")

```

```{r, echo = FALSE}
library(DESeq2)
```

The files are gigantic, I decide to run RScripts on the terminal

```{r, eval = FALSE}
library(data.table)
fwrite(rcstar_m,"rcstar_m.csv",row.names = FALSE)
fwrite(rcstar_f,"rcstar_f.csv",row.names = FALSE)
fwrite(rchisat_m,"rchisat_m.csv",row.names = FALSE)
fwrite(rchisat_f,"rchisat_f.csv",row.names = FALSE)
```



```bash
cd /athena/angsd/scratch/tof4003/project
mkdir -p Rscripts
cd Rscripts
nano makeDESeq.sh

#!/bin/bash
#SBATCH --partition=angsd_class
#SBATCH --account=angsd_class
#SBATCH --nodes=1
#SBATCH --job-name=feat_counts
#SBATCH --time=12:00:00 #HH/MM/SS
#SBATCH --mem=15G

cd /athena/angsd/scratch/tof4003/project/Rscripts
spack load -r /bxc56dm

Rscript DESeq2.R

nano DESeq2.R

```
in DESeq2.R, write in the Rscript below

```{r, eval = FALSE}
rcstar_m <- read.csv("rcstar_m.csv",header = TRUE)
rcstar_f<- read.csv("rcstar_f.csv",header = TRUE)
rchisat_m<- read.csv("rchisat_m.csv",header = TRUE)
rchisat_f<- read.csv("rchisat_f.csv",header = TRUE)
#BiocManager::install("DESeq2")
library(DESeq2)
#this function joins every element of 2 lists together to have a new list
#there are so many duplicate names, have to use a dictionary to give them small numbers
listjoin <- function(l1, l2,l3){
  new_d <- c("placeholder"=0)
  new_l <- list()
  for (i in 1: length(l1)){
    ns <- paste0(l1[i],"_",l2[i],l3[i])
    if(ns %in% names(new_d)){
      add_l<-paste0(ns,"_",new_d[ns])
      new_l <- append(add_l,new_l)
      new_d[ns] <- new_d[ns]+1
    }
    else{
      new_l <- append(ns, new_l)
      new_d[ns]<-1
    }
    
  }
  return(new_l)
}

rcstar_m$rows <- listjoin(rcstar_m$Geneid,rcstar_m$Start,rcstar_m$Strand)
row.names(rcstar_m) <- make.names(rcstar_m$Geneid)
rcstar_m <- rcstar_m[ ,-c(1:6)]
sampleinfo_starm <- DataFrame(condition = gsub("SRR+","",names(rcstar_m)),row.names = names(rcstar_m))

DESeq.starm <- DESeqDataSetFromMatrix(countData = as.matrix(rcstar_m),
                                   colData = sample_info_starm,
                                   design = ~ condition ) 

rcstar_f$rows <- listjoin(rcstar_f$Geneid,rcstar_f$Start,rcstar_f$Strand)
row.names(rcstar_f) <- make.names(rcstar_f$Geneid)
rcstar_f <- rcstar_f[ ,-c(1:6)]
sampleinfo_starf <- DataFrame(condition = gsub("SRR+","",names(rcstar_f)),row.names = names(rcstar_f))

DESeq.starf <- DESeqDataSetFromMatrix(countData = as.matrix(rcstar_f),
                                   colData = sample_info_starf,
                                   design = ~ condition ) 


rchisat_m$rows <- listjoin(rchisat_m$Geneid,rchisat_m$Start,rchisat_m$Strand)
row.names(rchisat_m) <- make.names(rchisat_m$Geneid)
rchisat_m <- rchisat_m[ ,-c(1:6)]
sampleinfo_hisatm <- DataFrame(condition = gsub("SRR+","",names(rchisat_m)),row.names = names(rchisat_m))

DESeq.hisatm <- DESeqDataSetFromMatrix(countData = as.matrix(rcstar_m),
                                   colData = sample_info_starm,
                                   design = ~ condition ) 

rchisat_f$rows <- listjoin(rchisat_f$Geneid,rchisat_f$Start,rchisat_f$Strand)
row.names(rchisat_f) <- make.names(rchisat_f$Geneid)
rchisat_f <- rchisat_f[ ,-c(1:6)]
sampleinfo_hisatf <- DataFrame(condition = gsub("SRR+","",names(rchisat_f)),row.names = names(rchisat_f))

DESeq.hisatf <- DESeqDataSetFromMatrix(countData = as.matrix(rchisat_f),
                                   colData = sample_info_hisatf,
                                   design = ~ condition ) 

save(DESeq.starm, file="DESeq.starm.RData")
save(DESeq.starf, file="DESeq.starf.RData")
save(DESeq.hisatm,file="DESeq.hisatm.RData")
save(DESeq.hisatf,file="DESeq.hisatf.RData")
```

Download and Upload

```bash
scp "C:/Users/tof4003/Desktop/WCM/W2022/CMPB 5004/Project/*.csv" tof4003@aphrodite.med.cornell.edu:/athena/angsd/scratch/tof4003/project/Rscripts 

sbatch makeDESeq.sh

```
