---
title: "Project Vignette"
author: "Phoebe Fei"
date: "3/26/2022"
output: 
  html_document:
    toc: true
    toc_float: true
    toc_depth: 3
---

```{css style_settings, echo = FALSE}
blockquote { 
    font-size: 14px; 
    font-style: italic;
}
h1{
    font-size: 22px;
}
h2{
    font-size: 20px;
}
h3{
    font-size: 18px;
}
h4{
    font-size: 16px;
}

```

**This is the main workflow .rmd that guides through the workflow of this project, some copied from the checkpoint rmd**

# 1. File Download and Preparation

### a. File Download 

The reference genomes for naked mole rats are found in Ensembl. I will use the female reference genome only, since the male genomes have unconventional naming for the genes. I will be testing on the liver RNAs, so there shouldn't be too many different gene names between male and female. 

For Female: http://ftp.ensembl.org/pub/current_gtf/heterocephalus_glaber_female/

The data is accessible from NCBI SRA/GEO dataset under project number PRJNA386015. The link to the SRA run selector is https://www.ncbi.nlm.nih.gov/Traces/study/?acc=PRJNA386015&o=acc_s%3Aa.  

i. Copy the files (reference and fastq files to my directory at /athena/angsd/scratch/tof4003/project)

I will copy all NMR liver files to the project

```bash
#download the metadata file (liver for all NMR) and put it into my directory
scp "C:/Users/tof4003/Desktop/WCM/W2022/CMPB 5004/Project/SraRunTable_liver.txt" tof4003@aphrodite.med.cornell.edu:/athena/angsd/scratch/tof4003/project

head -5 SraRunTable_liver.txt
```
#### Output
```
Run,Age,Assay Type,AvgSpotLen,Bases,BioProject,BioSample,Bytes,Center Name,Consent,DATASTORE filetype,DATASTORE provider,DATASTORE region,Experiment,GEO_Accession (exp),Instrument,LibraryLayout,LibrarySelection,LibrarySource,Organism,Platform,ReleaseDate,reproductive_status,Sample Name,sex,source_name,SRA Study,tissue,strain
SRR5517432,3.7 years,RNA-Seq,50,1143144100,PRJNA385850,SAMN06917366,753974373,GEO,public,"fastq,sra","gs,ncbi,s3","gs.US,ncbi.public,s3.us-east-1",SRX2790801,GSM2609133,Illumina HiSeq 2500,SINGLE,cDNA,TRANSCRIPTOMIC,Heterocephalus glaber,ILLUMINA,2018-06-14T00:00:00Z,breeder,GSM2609133,female,liver,SRP106650,liver,
SRR5517435,3.6 years,RNA-Seq,50,1263344450,PRJNA385850,SAMN06917363,850450852,GEO,public,"fastq,sra","gs,ncbi,s3","gs.US,ncbi.public,s3.us-east-1",SRX2790804,GSM2609136,Illumina HiSeq 2500,SINGLE,cDNA,TRANSCRIPTOMIC,Heterocephalus glaber,ILLUMINA,2018-06-14T00:00:00Z,breeder,GSM2609136,female,liver,SRP106650,liver,
SRR5517437,3.7 years,RNA-Seq,50,1109269000,PRJNA385850,SAMN06917361,731997749,GEO,public,"fastq,sra","gs,ncbi,s3","gs.US,ncbi.public,s3.us-east-1",SRX2790806,GSM2609138,Illumina HiSeq 2500,SINGLE,cDNA,TRANSCRIPTOMIC,Heterocephalus glaber,ILLUMINA,2018-06-14T00:00:00Z,breeder,GSM2609138,female,liver,SRP106650,liver,
SRR5517441,2.4 years,RNA-Seq,51,1253621004,PRJNA385850,SAMN06917357,846076900,GEO,public,"fastq,sra","gs,ncbi,s3","gs.US,ncbi.public,s3.us-east-1",SRX2790810,GSM2609142,Illumina HiSeq 2500,SINGLE,cDNA,TRANSCRIPTOMIC,Heterocephalus glaber,ILLUMINA,2018-06-14T00:00:00Z,non-breeder,GSM2609142,female,liver,SRP106650,liver,

```
The run is field 1, reproduction status is field 28, sex is field 30


```bash

#download data
awk -F "," '{ if($30 == "male") print $1","$28}' SraRunTable_liver.txt > liver_male.txt
awk -F "," '{ if($30 == "female") print $1","$28}' SraRunTable_liver.txt > liver_female.txt

mkdir -p lvr_male
mkdir -p lvr_female

#had to sbatch because this process took a long time
nano data_acc.sh
#! /bin/bash -l
#SBATCH --partition=angsd_class
#SBATCH --account=angsd_class
#SBATCH --nodes=1
#SBATCH --ntasks=1
#SBATCH --job-name=SRA_run
#SBATCH --time=15:00:00 #HH/MM/SS
#SBATCH --mem=10G

#load SRA toolkit
spack load sra-toolkit@2.8.2-1%gcc@6.3.0
cd /athena/angsd/scratch/tof4003/project
#clear the sra directory
rm -r /home/tof4003/ncbi
while read line; do
    echo $line
    acc=$(cut -d "," -f 1 <<< "$line")
    breed=$(cut -d "," -f 2 <<< "$line")
    prefetch $acc
    #change to fastq format
    fastq-dump /home/tof4003/ncbi/public/sra/$acc.sra -O /athena/angsd/scratch/tof4003/project/lvr_male
    #rename the file
    mv /athena/angsd/scratch/tof4003/project/lvr_male/$acc.fastq /athena/angsd/scratch/tof4003/project/lvr_male/$acc.m_$breed.fastq
done < /athena/angsd/scratch/tof4003/project/liver_male.txt
rm -r /home/tof4003/ncbi
while read line; do
    echo $line
    acc=$(cut -d "," -f 1 <<< "$line")
    breed=$(cut -d "," -f 2 <<< "$line")
    prefetch $acc
    #change to fastq format
    fastq-dump /home/tof4003/ncbi/public/sra/$acc.sra -O /athena/angsd/scratch/tof4003/project/lvr_female
    #rename the file
    mv /athena/angsd/scratch/tof4003/project/lvr_female/$acc.fastq /athena/angsd/scratch/tof4003/project/lvr_female/$acc.f_$breed.fastq
done < /athena/angsd/scratch/tof4003/project/liver_female.txt

sbatch data_acc.sh
#cleaning the temp files
rm -r /home/tof4003/ncbi

#download reference
mkdir -p NMR_ref

nano ref_d.sh

#! /bin/bash -l
#SBATCH --partition=angsd_class
#SBATCH --account=angsd_class
#SBATCH --nodes=1
#SBATCH --ntasks=1
#SBATCH --job-name=SRA_run
#SBATCH --time=15:00:00 #HH/MM/SS
#SBATCH --mem=10G

wget -O NMR_ref/femaleNMR.gtf.gz http://ftp.ensembl.org/pub/current_gtf/heterocephalus_glaber_female/Heterocephalus_glaber_female.HetGla_female_1.0.105.gtf.gz
#need to also download the reference fasta dna file
wget -O NMR_ref/NMR_female_dna.fa.gz http://ftp.ensembl.org/pub/current_fasta/heterocephalus_glaber_female/dna/Heterocephalus_glaber_female.HetGla_female_1.0.dna.toplevel.fa.gz

#unzip everything (for some reason directly specifying zcat to read .gz file with star returns that I have an extra space before the > in fasta, even though checking using head indicates no problem)
zcat /athena/angsd/scratch/tof4003/project/NMR_ref/femaleNMR.gtf.gz > /athena/angsd/scratch/tof4003/project/NMR_ref/femaleNMR.gtf
zcat /athena/angsd/scratch/tof4003/project/NMR_ref/NMR_female_dna.fa.gz > /athena/angsd/scratch/tof4003/project/NMR_ref/NMR_ref_female.fa

sbatch ref_d.sh

ls lvr_female
ls lvr_male
head -10 NMR_ref/femaleNMR.gtf
```
#### Output

lvr_female

```
SRR5517432.f_breeder.fastq  SRR5517441.f_non-breeder.fastq
SRR5517435.f_breeder.fastq  SRR5517445.f_non-breeder.fastq
SRR5517436.f_breeder.fastq  SRR5517446.f_non-breeder.fastq
SRR5517437.f_breeder.fastq  SRR5517449.f_non-breeder.fastq
SRR5517439.f_breeder.fastq  SRR5517450.f_non-breeder.fastq
SRR5517440.f_breeder.fastq

```
lvr_male
```
SRR5517433.m_breeder.fastq      SRR5517447.m_breeder.fastq
SRR5517434.m_breeder.fastq      SRR5517448.m_non-breeder.fastq
SRR5517438.m_breeder.fastq      SRR5517451.m_non-breeder.fastq
SRR5517442.m_non-breeder.fastq  SRR5517452.m_non-breeder.fastq
SRR5517443.m_breeder.fastq      SRR5517453.m_non-breeder.fastq
SRR5517454.m_breeder.fastq
SRR5517444.m_non-breeder.fastq

```

Features in the gtf file
```
#!genome-build HetGla_female_1.0
#!genome-version HetGla_female_1.0
#!genome-date 2012-02
#!genome-build-accession GCA_000247695.1
#!genebuild-last-updated 2017-07
JH602043.1	ensembl	gene	4190633	4194963	.	-	.	gene_id "ENSHGLG00000010267"; gene_version "1"; gene_name "ZMYND10"; gene_source "ensembl"; gene_biotype "protein_coding";
JH602043.1	ensembl	transcript	4190633	4194963	.	-	.	gene_id "ENSHGLG00000010267"; gene_version "1"; transcript_id "ENSHGLT00000014491"; transcript_version "1"; gene_name "ZMYND10"; gene_source "ensembl"; gene_biotype "protein_coding"; transcript_name "ZMYND10-201"; transcript_source "ensembl"; transcript_biotype "protein_coding";
JH602043.1	ensembl	exon	4194756	4194963	.	-	.	gene_id "ENSHGLG00000010267"; gene_version "1"; transcript_id "ENSHGLT00000014491"; transcript_version "1"; exon_number "1"; gene_name "ZMYND10"; gene_source "ensembl"; gene_biotype "protein_coding"; transcript_name "ZMYND10-201"; transcript_source "ensembl"; transcript_biotype "protein_coding"; exon_id "ENSHGLE00000106764"; exon_version "1";
JH602043.1	ensembl	CDS	4194756	4194847	.	-	0	gene_id "ENSHGLG00000010267"; gene_version "1"; transcript_id "ENSHGLT00000014491"; transcript_version "1"; exon_number "1"; gene_name "ZMYND10"; gene_source "ensembl"; gene_biotype "protein_coding"; transcript_name "ZMYND10-201"; transcript_source "ensembl"; transcript_biotype "protein_coding"; protein_id "ENSHGLP00000014347"; protein_version "1";
JH602043.1	ensembl	start_codon	4194845	4194847	.	-	0	gene_id "ENSHGLG00000010267"; gene_version "1"; transcript_id "ENSHGLT00000014491"; transcript_version "1"; exon_number "1"; gene_name "ZMYND10"; gene_source "ensembl"; gene_biotype "protein_coding"; transcript_name "ZMYND10-201"; transcript_source "ensembl"; transcript_biotype "protein_coding";
```

### b. creating Index

Create STAR index for NMR female, with editing the resource request to speed up as suggested

```bash

mkdir -p star_liver
nano star_liver/star_female_ref.sh

#! /bin/bash -l
#SBATCH --partition=angsd_class
#SBATCH --account=angsd_class
#SBATCH --nodes=1
#SBATCH #SBATCH --cpus-per-task=4 #try multiple threads with multiple cpu
#SBATCH --job-name=REf_gen_STAR
#SBATCH --time=48:00:00 #HH/MM/SS
#SBATCH --mem=100G

cd /athena/angsd/scratch/tof4003/project/NMR_ref
mkdir -p STAR_ref_female
#load STAR
spack load star@2.7.0e
STAR --runMode genomeGenerate \
--runThreadN 4 \
--genomeDir STAR_ref_female \
--genomeFastaFiles NMR_ref_female.fa \
--sjdbGTFfile femaleNMR.gtf \
--limitGenomeGenerateRAM 100326568150 \
--sjdbOverhang 99

sbatch star_liver/star_female_ref.sh
```


### c. Checking Quality of fastqc files

Running fastqc, then Multiqc on the fastq files

```bash
cd /athena/angsd/scratch/tof4003/project
#freeing some memories
nano gz.sh

#! /bin/bash -l
#SBATCH --partition=angsd_class
#SBATCH --account=angsd_class
#SBATCH --nodes=1
#SBATCH --ntasks=1
#SBATCH --job-name=zip_files
#SBATCH --time=15:00:00 #HH/MM/SS
#SBATCH --mem=8G

cd /athena/angsd/scratch/tof4003/project
gzip lvr_male/*.fastq lvr_male/*.fastq.gz
rm lvr_male/*.fastq
gzip lvr_female/*.fastq lvr_female/*fastq.gz
rm lvr_female/*.fastq

sbatch gz.sh

mkdir -p fastq_qc
cd fastq_qc
nano fastqc.sh 
#! /bin/bash -l
#SBATCH --partition=angsd_class
#SBATCH --account=angsd_class
#SBATCH --nodes=1
#SBATCH --job-name=fastqc
#SBATCH --time=12:00:00 #HH/MM/SS
#SBATCH --mem=8G

spack load fastqc
cd /athena/angsd/scratch/tof4003/project/lvr_male
for file in *.fastq.gz ; do
  echo $file
  fastqc $file -o /athena/angsd/scratch/tof4003/project/fastq_qc
done
cd /athena/angsd/scratch/tof4003/project/lvr_female
for file in *.fastq.gz ; do
  echo $file
  fastqc $file -o /athena/angsd/scratch/tof4003/project/fastq_qc
done
#then, run a multiqc
cd /athena/angsd/scratch/tof4003/project/fastq_qc
spack load -r py-multiqc
multiqc .

sbatch fastqc.sh

ls fastq_qc

-exit-
#file download in cmd
scp tof4003@aphrodite.med.cornell.edu:/athena/angsd/scratch/tof4003/project/fastq_qc/multiqc_report.html "/Users/phoebefei/Desktop/WCM/W2022/CMPB 5004/Project/ANGSD Project tof4003/QC htmls/fastqc_report.html"

````

#### Output

**Sequence Counts**
![](/Users/phoebefei/Desktop/WCM/W2022/CMPB 5004/Project/ANGSD Project tof4003/QC htmls/Sequence_Counts.png)

**GC Contents**
![](/Users/phoebefei/Desktop/WCM/W2022/CMPB 5004/Project/ANGSD Project tof4003/QC htmls/GC content.png)
**Duplication Level**
![](/Users/phoebefei/Desktop/WCM/W2022/CMPB 5004/Project/ANGSD Project tof4003/QC htmls/Duplication Level.png)

The quality, GC contents and duplication level look good - very likely, the adaptors are already trimmed. I will carry on without using trimgalore. 
There is a test on trimgalore with 1 sequence and shows that there is no much difference - see checkpoint1 html file.

# 2. Alignment

I will run STAR and Hisat2 - STAR is extremely slow but since it's the standard, I will run it. Hisat2 has generally stable performance.


### a. Hisat2

```bash
mkdir -p liver_hisat2
cd liver_hisat2

nano hisat2_script.sh
#! /bin/bash -l
#SBATCH --partition=angsd_class
#SBATCH --account=angsd_class
#SBATCH --nodes=1
#SBATCH --cpus-per-task=4 #try multiple threads with multiple cpu
#SBATCH --job-name=Hisat2_tof4003
#SBATCH --time=48:00:00 #HH/MM/SS
#SBATCH --mem=60G

#load modules
spack load hisat2
spack load samtools@1.9% gcc@6.3.0
cd /athena/angsd/scratch/tof4003/project/NMR_ref
#Index the genome
mkdir -p hisat2_index_NMR_female
cd hisat2_index_NMR_female
hisat2-build -p 16 /athena/angsd/scratch/tof4003/project/NMR_ref/NMR_ref_female.fa HISAT2_NMR_female
cd /athena/angsd/scratch/tof4003/project/lvr_male

#run hisat2 alignment: reads are in fastq files, threads = 2; files are not pair-end reads
mkdir -p /athena/angsd/scratch/tof4003/project/liver_hisat2/out_male
for file in *.fastq.gz; do
  echo "Begin $file"
  fv=$(cut -d "." -f 1 <<< "$file")
  suf=$(cut -d "." -f 2 <<< "$file")
  hisat2 -q -p 2 -x /athena/angsd/scratch/tof4003/project/NMR_ref/hisat2_index_NMR_female/HISAT2_NMR_female --summary-file /athena/angsd/scratch/tof4003/project/liver_hisat2/$fv.$suf.summary.txt -U $file -S /athena/angsd/scratch/tof4003/project/liver_hisat2/out_male/$fv.$suf.hisat2.sam
  #use samtools to covert to bam, sort& index
  #sort & covert to bam
  samtools view -b /athena/angsd/scratch/tof4003/project/liver_hisat2/out_male/$fv.$suf.hisat2.sam | samtools sort -o /athena/angsd/scratch/tof4003/project/liver_hisat2/out_male/$fv.$suf.hisat2.sorted.bam
  #index
  samtools index /athena/angsd/scratch/tof4003/project/liver_hisat2/out_male/$fv.$suf.hisat2.sorted.bam
  echo "Hisat2 for $fv.$suf done"
  #remove .sam files to free up space
  rm /athena/angsd/scratch/tof4003/project/liver_hisat2/out_male/$fv.$suf.hisat2.sam
done

#females
cd /athena/angsd/scratch/tof4003/project/lvr_female

mkdir -p /athena/angsd/scratch/tof4003/project/liver_hisat2/out_female
for file in *.fastq.gz; do
  echo "Begin $file"
  fv=$(cut -d "." -f 1 <<< "$file")
  suf=$(cut -d "." -f 2 <<< "$file")
  hisat2 -q -p 2 -x /athena/angsd/scratch/tof4003/project/NMR_ref/hisat2_index_NMR_female/HISAT2_NMR_female --summary-file /athena/angsd/scratch/tof4003/project/liver_hisat2/$fv.$suf.summary.txt -U $file -S /athena/angsd/scratch/tof4003/project/liver_hisat2/out_female/$fv.$suf.hisat2.sam
  #use samtools to covert to bam, sort& index
  #sort & covert to bam
  samtools view -b /athena/angsd/scratch/tof4003/project/liver_hisat2/out_female/$fv.$suf.hisat2.sam | samtools sort -o /athena/angsd/scratch/tof4003/project/liver_hisat2/out_female/$fv.$suf.hisat2.sorted.bam
  #index
  samtools index /athena/angsd/scratch/tof4003/project/liver_hisat2/out_female/$fv.$suf.hisat2.sorted.bam
  echo "Hisat2 for $fv.$suf done"
  #remove .sam files to free up space
  rm /athena/angsd/scratch/tof4003/project/liver_hisat2/out_female/$fv.$suf.hisat2.sam
done

sbatch hisat2_script.sh

```

### b. STAR

```bash
cd /athena/angsd/scratch/tof4003/project/star_liver

nano star_align.sh
#star actually gives core-dump issue for male genomes if not enough core is reserved. 

#! /bin/bash -l
#SBATCH --partition=angsd_class
#SBATCH --account=angsd_class
#SBATCH --nodes=1
#SBATCH --cpus-per-task=8
#SBATCH --job-name=STAR_align
#SBATCH --time=40:00:00 #HH/MM/SS
#SBATCH --mem=90G

#load modules
spack load star@2.7.0e
spack load samtools@1.9% gcc@6.3.0
cd /athena/angsd/scratch/tof4003/project/star_liver
#running STAR
for file in /athena/angsd/scratch/tof4003/project/lvr_male/*.fastq.gz; do
  echo "Begin $file"
  fv=$(cut -d "/" -f 8 <<< "$file" | cut -d "." -f 1 )
  suf=$(cut -d "/" -f 8 <<< "$file" | cut -d "." -f 2)
  STAR --runMode alignReads \
  --runThreadN 8 \
  --genomeDir /athena/angsd/scratch/tof4003/project/NMR_ref/STAR_ref_female \
  --readFilesCommand zcat \
  --readFilesIn $file \
  --outFileNamePrefix $fv.$suf.star. \
  --outSAMtype BAM SortedByCoordinate
  #use samtools to index
  #index
  samtools index $fv.$suf.star.Aligned.sortedByCoord.out.bam
  echo "star $fv.$suf done"
done
#females
for file in /athena/angsd/scratch/tof4003/project/lvr_female/*.fastq.gz; do
  echo "Begin $file"
  fv=$(cut -d "/" -f 8 <<< "$file" | cut -d "." -f 1 )
  suf=$(cut -d "/" -f 8 <<< "$file" | cut -d "." -f 2)
  STAR --runMode alignReads \
  --runThreadN 4 \
  --genomeDir /athena/angsd/scratch/tof4003/project/NMR_ref/STAR_ref_female \
  --readFilesCommand zcat \
  --readFilesIn $file \
  --outFileNamePrefix $fv.$suf.star. \
  --outSAMtype BAM SortedByCoordinate
  #use samtools to index
  #index
  samtools index $fv.$suf.star.Aligned.sortedByCoord.out.bam
  echo "star $fv.$suf done"
done

sbatch star_align.sh
```

# 3.  Quality Check

### a. clearing the files

```bash
cd /athena/angsd/scratch/tof4003/project

#separate out the female and female files for star
mkdir -p star_liver/female_out
mkdir -p star_liver/male_out

mv star_liver/*f_* star_liver/female_out
mv star_liver/*m_* star_liver/male_out
```


### b. Multiqc on the Output Reports


```bash
cd /athena/angsd/scratch/tof4003/project
mkdir -p alignment_repo
cd alignment_repo
mkdir -p star
mkdir -p hisat2
#copy the log files to directories 
cp /athena/angsd/scratch/tof4003/project/star_liver/male_out/*.Log.final.out star
cp /athena/angsd/scratch/tof4003/project/star_liver/female_out/*.Log.final.out star
cp /athena/angsd/scratch/tof4003/project/liver_hisat2/*.txt hisat2

nano align_multiqc.sh

#! /bin/bash -l
#SBATCH --partition=angsd_class
#SBATCH --account=angsd_class
#SBATCH --nodes=1
#SBATCH --job-name=bam_fastqc
#SBATCH --time=12:00:00 #HH/MM/SS
#SBATCH --mem=14G

cd /athena/angsd/scratch/tof4003/project/alignment_repo
#run multiqc
spack load -r py-multiqc
cd star
multiqc .
cd /athena/angsd/scratch/tof4003/project/alignment_repo/hisat2
multiqc .

sbatch align_multiqc.sh

#file download in cmd
scp tof4003@aphrodite.med.cornell.edu:/athena/angsd/scratch/tof4003/project/alignment_repo/hisat2/multiqc_report.html "/Users/phoebefei/Desktop/WCM/W2022/CMPB 5004/Project/ANGSD Project tof4003/QC htmls/hisat2_qc.html"

scp tof4003@aphrodite.med.cornell.edu:/athena/angsd/scratch/tof4003/project/alignment_repo/star/multiqc_report.html "/Users/phoebefei/Desktop/WCM/W2022/CMPB 5004/Project/ANGSD Project tof4003/QC htmls/star_qc.html"
```

#### Output of Some Selected Features

**STAR Alignment Scores**
![](/Users/phoebefei/Desktop/WCM/W2022/CMPB 5004/Project/ANGSD Project tof4003/QC htmls/star_alignment_plot.png)
### c. RSeQC/QoRTs on bam file & gtf

need to download .bed file
```bash

```


# 4. Feature Counts

```bash

cd /athena/angsd/scratch/tof4003/project

#will generate count tables for star and hisat2 separately
mkdir -p featureCounts
cd featureCounts

nano featCounts.sh

#! /bin/bash -l
#SBATCH --partition=angsd_class
#SBATCH --account=angsd_class
#SBATCH --nodes=1
#SBATCH --job-name=feat_counts
#SBATCH --time=12:00:00 #HH/MM/SS
#SBATCH --mem=15G

# for the -O I am not going to include the "-O" reads can be assigned to more than 1 matched exon (can be counted more than once) - so if reads matched more than one exon, all of them will be counted. As discussed in the hw, the ambiguous reads are most likely UTR's so I actually don't need them
cd /athena/angsd/scratch/tof4003/project
spack load subread
# for the -O I am not going to include the "-O" reads can be assigned to more than 1 matched exon (can be counted more than once) - so if reads matched more than one exon, all of them will be counted. As discussed in the hw, the ambiguous reads are most likely UTR's so I actually don't need them
# male and female needs to be done separately (2 references)
# deleting the -f flag because we actually want to be mapped to genes
# because the females gtf files are nicely annotated, I can include --extraAttributes for gene_biotype & gene_name

featureCounts -a NMR_ref/femaleNMR.gtf \
-o featureCounts/star_featCounts_maleNMR.txt \
-t exon \
-g gene_id \
--extraAttributes gene_name,gene_biotype \
-G NMR_ref/NMR_ref_female.fa \
star_liver/male_out/*.bam

featureCounts -a NMR_ref/femaleNMR.gtf \
-o featureCounts/star_featCounts_femaleNMR.txt \
-t exon \
-g gene_id \
--extraAttributes gene_name,gene_biotype \
-G NMR_ref/NMR_ref_female.fa \
star_liver/female_out/*.bam

featureCounts -a NMR_ref/femaleNMR.gtf \
-o featureCounts/hisat2_featCounts_maleNMR.txt \
-t exon \
-g gene_id \
--extraAttributes gene_name,gene_biotype \
-G NMR_ref/NMR_ref_female.fa \
liver_hisat2/out_male/*.bam

featureCounts -a NMR_ref/femaleNMR.gtf \
-o featureCounts/hisat2_featCounts_femaleNMR.txt \
-t exon \
-g gene_id \
--extraAttributes gene_name,gene_biotype \
-G NMR_ref/NMR_ref_female.fa \
liver_hisat2/out_female/*.bam

sbatch featCounts.sh
cat *.out


--------------file download

scp tof4003@aphrodite.med.cornell.edu:/athena/angsd/scratch/tof4003/project/featureCounts/*.txt "C:\Users\tof4003\Desktop\WCM\W2022\CMPB 5004\Project\ANGSD Project tof4003"
scp tof4003@aphrodite.med.cornell.edu:/athena/angsd/scratch/tof4003/project/featureCounts/*.txt.summary "C:\Users\tof4003\Desktop\WCM\W2022\CMPB 5004\Project\ANGSD Project tof4003"

```

#### Output

showing feature counts for 1
```

```

# 5. Analyze Count Data

### a. Import and process the summary Table
```{r}
library(ggplot2)
library(magrittr)
direct <- '/Users/phoebefei/Desktop/WCM/W2022/CMPB 5004/Project/ANGSD Project tof4003/'
#hisat2 males & females
rchisat2f <-paste0(direct,"hisat2_featCounts_femaleNMR.txt.summary") %>%
  read.table(., header = TRUE)

rchisat2m <-paste0(direct,"hisat2_featCounts_maleNMR.txt.summary") %>%
  read.table(., header = TRUE)

#star males & females
rcstarf <-paste0(direct,"star_featCounts_femaleNMR.txt.summary") %>%
  read.table(., header = TRUE)

rcstarm <-paste0(direct,"star_featCounts_maleNMR.txt.summary") %>%
  read.table(., header = TRUE)

#hisat2f has an extra column by mistake
#names(rchisat2f)
rchisat2f<- subset(rchisat2f, select = -`liver_hisat2.out_female...fastq.hisat2.sorted.bam`)
```
Clean the sample names

```{r}
library(stringr)
#reassign sample names
nameliststarm <- str_extract(names(rcstarm), "SRR[0-9]+.(m|f)_[a-z]+")
namelisthisat2m <- str_extract(names(rchisat2m), "SRR[0-9]+.(m|f)_[a-z]+")
nameliststarf <- str_extract(names(rcstarf), "SRR[0-9]+.(m|f)_[a-z]+")
namelisthisat2f <- str_extract(names(rchisat2f), "SRR[0-9]+.(m|f)_[a-z]+")
```
```{r}
#don't want the status to be modified
names(rcstarm)[2:length(rcstarm)] <- nameliststarm[2:length(nameliststarm)]
names(rcstarf)[2:length(rcstarf)] <- nameliststarf[2:length(nameliststarf)]
names(rchisat2m)[2:length(rchisat2m)] <- namelisthisat2m[2:length(namelisthisat2m)]

names(rchisat2f)[2:length(rchisat2f)] <- namelisthisat2f[2:length(namelisthisat2f)]
str(rchisat2f)
str(rchisat2m)
str(rcstarf)
str(rcstarm)
```

Plotting
```{r}
library(patchwork)
library(reshape2)
#STAR M
melt_starm <- melt(rcstarm, id.vars = "Status")
names(melt_starm)[3] <- "Counts"
names(melt_starm)[2] <- "Sample"
#remove 0 instances
melt_starm <- subset(melt_starm, Counts!=0)
#add a row saying identifying this data
melt_starm["Type"] <- rep("Feature Count on Male using STAR",nrow(melt_starm))
#STAR F
melt_starf <- melt(rcstarf, id.vars = "Status")
names(melt_starf)[3] <- "Counts"
names(melt_starf)[2] <- "Sample"
melt_starf <- subset(melt_starf, Counts!=0)
melt_starf["Type"] <- rep("Feature Count on Female using STAR",nrow(melt_starf))

#Hisat2 M
melt_hisat2m <- melt(rchisat2m, id.vars = "Status")
names(melt_hisat2m)[3] <- "Counts"
names(melt_hisat2m)[2] <- "Sample"
melt_hisat2m <- subset(melt_hisat2m, Counts!=0)
melt_hisat2m["Type"] <- rep("Feature Count on Male using Hisat2",nrow(melt_hisat2m))

#Hisat2 F
melt_hisat2f <- melt(rchisat2f, id.vars = "Status")
names(melt_hisat2f)[3] <- "Counts"
names(melt_hisat2f)[2] <- "Sample"
melt_hisat2f <- subset(melt_hisat2f, Counts!=0)
melt_hisat2f["Type"] <- rep("Feature Count on Male using Hisat2",nrow(melt_hisat2f))
```
Male Alignment : STAR vs. Hisat2

```{r}
melt_m <- rbind(melt_starm,melt_hisat2m)
melt_f <- rbind(melt_starf,melt_hisat2f)
```

```{r fig.width = 10, fig.height = 8}
#size figure size
#knitr::opts_chunk$set(fig.width=unit(30,"cm"), fig.height=unit(50,"cm"))
#set texts size
theme_set(theme(legend.position = "bottom", legend.key.size = unit(1, 'cm'),axis.text=element_text(size=14), axis.title=element_text(size=14)))
#bar plot the combined graph
plotm <- ggplot(data = melt_m,aes(x = Counts, y = Sample))+geom_bar(stat = 'identity', aes(fill = Status),position = "dodge")+facet_wrap(Type~., nrow =2, strip.position = "right")+ggtitle("Plot Based on Summary .txt Files")
plotm

```

```{r fig.width = 10, fig.height = 8}
#size figure size
#knitr::opts_chunk$set(fig.width=unit(30,"cm"), fig.height=unit(70,"cm"))
#set texts size
theme_set(theme(legend.position = "bottom", legend.key.size = unit(1, 'cm'),axis.text=element_text(size=14), axis.title=element_text(size=14)))
#bar plot the combined graph
plotf <- ggplot(data = melt_f,aes(x = Counts, y = Sample))+geom_bar(stat = 'identity', aes(fill = Status),width = 1,position = "dodge")+facet_wrap(Type~., nrow =2, strip.position = "right")+ggtitle("Plot Based on Summary .txt Files")
plotf


```

### b. create DESeq2 Objects

Import readcounts and assign sample names
```{r}
rchisat_f <-paste0(direct,"hisat2_featCounts_femaleNMR.txt") %>% read.table(., header = TRUE)
rchisat_m <- paste0(direct,"hisat2_featCounts_maleNMR.txt") %>% read.table(., header = TRUE)
rcstar_f <-paste0(direct,"star_featCounts_femaleNMR.txt") %>% read.table(., header = TRUE)
rcstar_m <- paste0(direct,"star_featCounts_maleNMR.txt") %>% read.table(., header = TRUE)
```
```{r}
#clean hisat2f extra column
#names(rchisat_f)
rchisat_f<- subset(rchisat_f, select = -`liver_hisat2.out_female...fastq.hisat2.sorted.bam`)

```
```{r}
#don't want the status to be modified
names(rcstar_m)[7:length(rcstar_m)] <- nameliststarm[2:length(nameliststarm)]
names(rcstar_f)[7:length(rcstar_f)] <- nameliststarf[2:length(nameliststarf)]
names(rchisat_m)[7:length(rchisat_m)] <- namelisthisat2m[2:length(namelisthisat2m)]
names(rchisat_f)[7:length(rchisat_f)] <- namelisthisat2f[2:length(namelisthisat2f)]

head(rcstar_m)
head(rcstar_f)
head(rchisat_m)
head(rchisat_f)
```

Merge male and female data

```{r}
#Create rownames
#Star Males
row.names(rcstar_m) <- make.names(rcstar_m$Geneid)
rcstar_m <- rcstar_m[ ,-c(1:6)]


# Star Females
row.names(rcstar_f) <- make.names(rcstar_f$Geneid)
rcstar_f <- rcstar_f[ ,-c(1:6)]


# Hisat2 Males
row.names(rchisat_m) <- make.names(rchisat_m$Geneid)
rchisat_m <- rchisat_m[ ,-c(1:6)]

#Hisat2 Females
row.names(rchisat_f) <- make.names(rchisat_f$Geneid)
rchisat_f <- rchisat_f[ ,-c(1:6)]


```

```{r}
#merge & only keep same gene name

rcstar <- merge(rcstar_m, rcstar_f,by=0, all = TRUE)
rcstar[is.na(rcstar)] <- 0
rchisat <- merge(rchisat_f, rchisat_m,by=0, all = TRUE)
rchisat[is.na(rchisat)] <- 0
row.names(rcstar) <- make.names(rcstar$Row.names)
rcstar <- rcstar[, -c(1)]
row.names(rchisat) <- make.names(rchisat$Row.names)
rchisat <- rchisat[, -c(1)]
```

```{r}
head(row.names(rchisat_f))
head(row.names(rchisat_m))

```

Create sex & breed condition

```{r}
star_level <- gsub("SRR[0-9]+.","",names(rcstar))
star_sex <- gsub("_.+","",star_level)
star_breed <- gsub("(m|f)_","",star_level)
hisat_level <- gsub("SRR[0-9]+.","",names(rchisat))
hisat_sex <- gsub("_.+","",hisat_level)
hisat_breed <- gsub("(m|f)_","",hisat_level)


```

```{r}
sampleinfo_hisat <- DataFrame(breed = hisat_breed, sex = hisat_sex, row.names = names(rchisat))
sampleinfo_star <- DataFrame(breed = star_breed, sex = star_sex ,row.names = names(rcstar))

```

Create DESeq2 Objects
```{r, eval = FALSE}
BiocManager::install("DESeq2")

```

```{r, echo = FALSE}
library("DESeq2")
DESeq.star <- DESeqDataSetFromMatrix(countData = as.matrix(rcstar),
                                   colData = sampleinfo_star,
                                   design = ~ breed + sex ) 


DESeq.hisat <- DESeqDataSetFromMatrix(countData = as.matrix(rchisat),
                                   colData = sampleinfo_hisat,
                                   design = ~ breed + sex ) 

```

### c. Access Read Counts

Cleaning the data
```{r}
colSums(counts(DESeq.star)) %>% barplot(las=2)
```
```{r}
dim(DESeq.star)
keep_star <- rowSums(counts(DESeq.star)) > 0
DESeq.star <- DESeq.star[keep_star,]
dim(DESeq.star)
```

Calculate Size Factors

```{r}
# define a function to calculate the geometric mean
gm_mean <- function(x, na.rm=TRUE){ exp(sum(log(x[x > 0]), na.rm=na.rm) / length(x)) }

star_pseudo_ref <- counts(DESeq.star) %>% apply(., 1, gm_mean)
star_ref_ratio <- counts(DESeq.star) %>% apply(., 2, function(cts){ cts/star_pseudo_ref})

counts(DESeq.star)[1,]/star_pseudo_ref[1]
star_ref_ratio[1,]

apply(star_ref_ratio,2,median)
```

```{r}
#gives error that gene contains zeros, temporarily add 1 to everything
DESeq.star <- estimateSizeFactors(DESeq.star, type = "poscounts")
plot(sizeFactors(DESeq.star), colSums(counts(DESeq.star)), ylab = "library sizes", xlab = "size factors", cex = 1)
```

Normalization and Plotting
```{r}
par(mfrow=c(1,2))

counts.star_norm <- counts(DESeq.star, normalized = TRUE)

boxplot(counts.star_norm, main = "SF normalized", las = 2, cex = .6)
boxplot(counts(DESeq.star), main = "Read counts only", las = 2, cex = .6)
```
boxplot of log2 transformation

```{r}
par(mfrow=c(1,2))

boxplot(log2(counts(DESeq.star)+1), notch=FALSE, main = "Non-normalized read counts", ylab="log2(read counts)", cex = .6, las = 2)

boxplot(log2(counts(DESeq.star, normalize= TRUE)+1), notch=FALSE, main = "Size-factor-normalized read counts", ylab="log2(read counts)", cex = .6, las = 2)


```

### d. Properties of the Read Count Data

2. run multiQC on star log file and hisat2 slurm file (rename the file)
3. RSeQC/QoRTs on bam file & gtf (instead of bamqc)
4. Save to rowData: gene biotype, gene name
5. saveRows when saving the results

